--- openjpeg-1.5.1/applications/codec/windirent.h.orig	2013-12-07 20:29:59.004295373 +0000
+++ openjpeg-1.5.1/applications/codec/windirent.h	2013-12-07 20:30:37.760707488 +0000
@@ -124,7 +124,7 @@
 #   error "not implemented"
 # elif defined(__sgi)                          /* Silicon Graphics */
 #   define HAVE_DIRENT_H
-# elif defined(sun) || defined(__sun)           /* Sun Solaris */
+# elif defined(__sun) || defined(__sun__)      /* Sun Solaris */
 #   define HAVE_DIRENT_H
 # elif defined(__FreeBSD__)                    /* FreeBSD */
 #   define HAVE_DIRENT_H
