--- postgresql-9.4.0/configure.in.~1~	2014-12-16 01:07:34.000000000 +0000
+++ postgresql-9.4.0/configure.in	2014-12-19 20:47:05.651199094 +0000
@@ -1022,13 +1022,6 @@
 #endif
 ])
 
-# On OpenBSD, test for sys/ucred.h will fail unless sys/param.h
-# is included first.
-AC_CHECK_HEADERS(sys/ucred.h, [], [],
-[AC_INCLUDES_DEFAULT
-#include <sys/param.h>
-])
-
 # At least on IRIX, test for netinet/tcp.h will fail unless
 # netinet/in.h is included first.
 AC_CHECK_HEADERS(netinet/in.h)
