--- yamdi-1.9/yamdi.c.orig	2012-12-09 17:15:00.000000000 +0000
+++ yamdi-1.9/yamdi.c	2013-01-28 16:04:46.545819197 +0000
@@ -2304,7 +2304,7 @@
 		fprintf(fp, "<times>\n");
 
 		for(i = 0; i < flv->keyframes.nkeyframes; i++)
-			fprintf(fp, "<value id=\"%" PRIu64 "\">%.2f</value>\n", (uint64_t)i, (double)flv->keyframes.keyframetimestamps[i] / 1000.0);
+			fprintf(fp, "<value id=\"%" PRIu64 "\">%.3f</value>\n", (uint64_t)i, (double)flv->keyframes.keyframetimestamps[i] / 1000.0);
 
 		fprintf(fp, "</times>\n");
 		fprintf(fp, "<filepositions>\n");
