--- giflib-5.1.4/doc/Makefile.in.orig	2017-01-05 14:25:42.867349667 +0000
+++ giflib-5.1.4/doc/Makefile.in	2016-04-02 15:36:45.000000000 +0000
@@ -609,18 +609,18 @@
 
 
 .xml.html:
-	xmlto xhtml-nochunks $<
+#	xmlto xhtml-nochunks $<
 
 .xml.1:
-	xmlto man $<
+#	xmlto man $<
 
 .xml.txt:
-	xmlto txt $<
+#	xmlto txt $<
 
 .asc.html:
 	asciidoc $<
 
-all: allhtml manpages
+all: manpages
 
 # Philosophical choice: the website gets the internal manual pages
 allhtml: $(XMLALL:.xml=.html) $(XMLINTERNAL:.xml=.html)
