--- a/configure.orig	2012-04-02 14:52:42.719936473 +0000
+++ b/configure	2012-04-02 14:53:24.687545529 +0000
@@ -15235,7 +15235,7 @@
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking $XAPIAN_CONFIG works" >&5
 $as_echo_n "checking $XAPIAN_CONFIG works... " >&6; }
 
-        if (exec >&5 2>&5 ; $XAPIAN_CONFIG --ltlibs --cxxflags; exit $?) then
+        if (exec >&5 2>&5 ; $XAPIAN_CONFIG --libs --cxxflags; exit $?) then
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
     else
@@ -15250,12 +15250,12 @@
 	as_fn_error $? "\`$XAPIAN_CONFIG' not executable, aborting" "$LINENO" 5
 	;;
       esac
-      as_fn_error $? "\`$XAPIAN_CONFIG --ltlibs --cxxflags' doesn't work, aborting" "$LINENO" 5
+      as_fn_error $? "\`$XAPIAN_CONFIG --libs --cxxflags' doesn't work, aborting" "$LINENO" 5
     fi
 
     XAPIAN_VERSION=`$XAPIAN_CONFIG --version|sed 's/.* //;s/_svn[0-9]*$//'`
     XAPIAN_CXXFLAGS=`$XAPIAN_CONFIG --cxxflags`
-    XAPIAN_LIBS=`$XAPIAN_CONFIG --ltlibs`
+    XAPIAN_LIBS=`$XAPIAN_CONFIG --libs`
     :
   fi
 
