--- sudo-1.8.1p2/plugins/sudoers/ins_csops.h.orig	2011-06-28 16:01:26.322432688 +0100
+++ sudo-1.8.1p2/plugins/sudoers/ins_csops.h	2011-06-28 16:06:25.163205771 +0100
@@ -23,17 +23,5 @@
      */
 
     "Maybe if you used more than just two fingers...",
-    "BOB says:  You seem to have forgotten your passwd, enter another!",
-    "stty: unknown mode: doofus",
-    "I can't hear you -- I'm using the scrambler.",
-    "The more you drive -- the dumber you get.",
-#ifdef PC_INSULTS
-    "Listen, broccoli brains, I don't have time to listen to this trash.",
-#else
-    "Listen, burrito brains, I don't have time to listen to this trash.",
-#endif
-    "I've seen penguins that can type better than that.",
-    "Have you considered trying to match wits with a rutabaga?",
-    "You speak an infinite deal of nothing",
 
 #endif /* _SUDO_INS_CSOPS_H */
