Index: gmp-5.1.2/gmp-h.in
===================================================================
--- gmp-5.1.2/gmp-h.in.orig
+++ gmp-5.1.2/gmp-h.in
@@ -29,9 +29,18 @@ along with the GNU MP Library.  If not,
 
 /* Instantiated by configure. */
 #if ! defined (__GMP_WITHIN_CONFIGURE)
+#if defined(__amd64) || defined(__sparcv9)
+#define __GMP_BITS_PER_MP_LIMB 64
+#else
+#define __GMP_BITS_PER_MP_LIMB 32
+#endif
 #define __GMP_HAVE_HOST_CPU_FAMILY_power   @HAVE_HOST_CPU_FAMILY_power@
 #define __GMP_HAVE_HOST_CPU_FAMILY_powerpc @HAVE_HOST_CPU_FAMILY_powerpc@
-#define GMP_LIMB_BITS                      @GMP_LIMB_BITS@
+#if defined(__amd64) || defined(__sparcv9)
+#define GMP_LIMB_BITS 64
+#else
+#define GMP_LIMB_BITS 32
+#endif
 #define GMP_NAIL_BITS                      @GMP_NAIL_BITS@
 #endif
 #define GMP_NUMB_BITS     (GMP_LIMB_BITS - GMP_NAIL_BITS)
