--- mutt-1.9.5/configure.orig	2018-04-15 20:15:27.986338360 +0000
+++ mutt-1.9.5/configure	2018-04-15 20:16:46.527214183 +0000
@@ -9850,8 +9850,8 @@
 
   if test "$KRB5CFGPATH" != "none"
   then
-    GSSAPI_CFLAGS="`$KRB5CFGPATH --cflags gssapi`"
-    GSSAPI_LIBS="`$KRB5CFGPATH --libs gssapi`"
+    GSSAPI_CFLAGS="`$KRB5CFGPATH --cflags`"
+    GSSAPI_LIBS="`$KRB5CFGPATH --libs` -lgss"
     case "`$KRB5CFGPATH --version`" in
       "Kerberos 5 "*)	GSSAPI_IMPL="MIT";;
       ?eimdal*)		GSSAPI_IMPL="Heimdal";;
--- mutt-1.5.22/imap/auth_gss.c.~1~	2010-02-21 22:10:41.000000000 +0000
+++ mutt-1.5.22/imap/auth_gss.c	2013-12-30 21:57:15.242624905 +0000
@@ -33,7 +33,8 @@
 #  define gss_nt_service_name GSS_C_NT_HOSTBASED_SERVICE
 #else
 #  include <gssapi/gssapi.h>
-#  include <gssapi/gssapi_generic.h>
+#  include <gssapi/gssapi_ext.h>
+#  define gss_nt_service_name GSS_C_NT_HOSTBASED_SERVICE
 #endif
 
 #define GSS_BUFSIZE 8192
