--- gpac/configure.~1~	2013-09-22 20:47:42.000000000 +0100
+++ gpac/configure	2013-09-23 13:39:35.992985899 +0100
@@ -57,6 +57,7 @@
 verbose="no"
 xulsdk_path="/usr/lib/xulrunner/sdk/include"
 xul_flags=""
+bindir="bin"
 libdir="lib"
 
 #GPAC module config
@@ -365,6 +366,7 @@
         case `uname -s` in
             SunOS)
                 canon_arch=`isainfo -n`
+                is_64="yes"
                 ;;
             *)
                 canon_arch="`$cc -dumpmachine | sed -e 's,\([^-]*\)-.*,\1,'`"
@@ -502,6 +504,7 @@
         fi
         if test "$is_64" = "yes"; then
             if test "$cpu" = "x86_64"; then
+                bindir=bin/amd64
                 libdir=lib/amd64
             elif test "$cpu" = "sparc"; then
                 libdir=lib/sparcv9
@@ -1274,36 +1277,16 @@
 #look for FFMPEG support
 
 ffmpeg_cflags=""
-ffmpeg_lflags="-lz -lavcodec -lavformat -lavutil -lswscale -lavdevice"
+ffmpeg_lflags="-lz -lavcodec -lavformat -lavutil -lswscale"
 if test "$cross_prefix" = "" -a "$pkg_config" != "no"; then
-  if $pkg_config --exists libavcodec libavformat libswscale libavdevice libavutil 2> /dev/null  ; then
-    ffmpeg_cflags=`$pkg_config --cflags libavcodec libavformat libswscale libavutil libavdevice`
-    ffmpeg_lflags=`$pkg_config --libs libavcodec libavformat libswscale libavutil libavdevice`
+  if $pkg_config --exists libavcodec libavformat libswscale libavutil 2> /dev/null  ; then
+    ffmpeg_cflags=`$pkg_config --cflags libavcodec libavformat libswscale libavutil`
+    ffmpeg_lflags=`$pkg_config --libs libavcodec libavformat libswscale libavutil`
     has_ffmpeg="system"
   fi
 fi
 
-cat > $TMPC << EOF
-#include <libavcodec/avcodec.h>
-int main(void) {return 0;}
-EOF
-
-if $cc -o $TMPO $TMPC $ffmpeg_cflags $ffmpeg_lflags $LDFLAGS 2> /dev/null  ; then
-    old_ffmpeg_inc="no"
-else
-    old_ffmpeg_inc="yes"
-
-    cat > $TMPC << EOF
-#include <ffmpeg/avcodec.h>
-int main(void) {
-    AVFrame *f1, *f2;
-    f2 = av_frame_clone(f1);
-    AV_CODEC_ID_H264;
-    return 0;
-}
-EOF
-
-fi
+old_ffmpeg_inc="no"
 
 if test "$cross_prefix" = "" ; then 
     if $cc -o $TMPO $TMPC -I$prefix/include -L$prefix/lib $ffmpeg_lflags $LDFLAGS 2> /dev/null  ; then
@@ -2724,6 +2707,7 @@
 echo "CPPFLAGS=$CPPFLAGS" >> config.mak
 echo "LDFLAGS=$LDFLAGS" >> config.mak
 echo "SHFLAGS=$SHFLAGS" >> config.mak
+echo "bindir=$bindir" >> config.mak
 echo "libdir=$libdir" >> config.mak
 
 echo "STATIC_MODULES=$static_modules" >> config.mak
