--- gpac-2011-07-19/configure.~1~	2011-07-19 11:11:41.000000000 +0100
+++ gpac-2011-07-19/configure	2011-07-19 16:14:38.228266091 +0100
@@ -155,7 +155,7 @@
 CPPFLAGS=""
 GPAC_SH_FLAGS=-lpthread
 DYN_LIB_SUFFIX="so"
-X11_PATH="/usr/X11R6"
+X11_PATH="/ec"
 OSS_CFLAGS=""
 OSS_LDFLAGS=""
 INSTFLAGS=""
@@ -303,6 +303,7 @@
         case `uname -s` in
             SunOS)
                 canon_arch=`isainfo -n`
+                is_64="yes"
                 ;;
             *)
                 canon_arch="`$cc -dumpmachine | sed -e 's,\([^-]*\)-.*,\1,'`"
@@ -2068,6 +2069,14 @@
 echo "SHFLAGS=$SHFLAGS" >> config.mak
 echo "libdir=$libdir" >> config.mak
 
+# Setting bindir
+bindir=bin
+if [ "$is_64" = "yes" -a "$sunos" = "yes" ] ; then
+    bindir=bin/amd64
+fi
+echo "bindir=$bindir" >> config.mak
+  
+
 
 #for cross-compilation
 if test "$cross_prefix"  != "" ; then
