--- gpac-fc32cebbe682a47804c8bf9eba6d0d4837aae88f/configure.orig	2014-11-28 20:39:47.510650434 +0000
+++ gpac-fc32cebbe682a47804c8bf9eba6d0d4837aae88f/configure	2014-11-28 20:45:27.449518425 +0000
@@ -55,6 +55,7 @@
 verbose="no"
 xulsdk_path="/usr/lib/xulrunner/sdk/include"
 xul_flags=""
+bindir="bin"
 libdir="lib"
 
 #GPAC module config
@@ -373,6 +374,7 @@
         case `uname -s` in
             SunOS)
                 canon_arch=`isainfo -n`
+                is_64="yes"
                 ;;
             Darwin)
                 canon_arch="x86_64"
@@ -514,6 +516,7 @@
         fi
         if test "$is_64" = "yes"; then
             if test "$cpu" = "x86_64"; then
+                bindir=bin/amd64
                 libdir=lib/amd64
             elif test "$cpu" = "sparc"; then
                 libdir=lib/sparcv9
@@ -1302,29 +1305,7 @@
   fi
 fi
 
-cat > $TMPC << EOF
-#include <libavcodec/avcodec.h>
-int main(void) {
-    return 0;
-}
-EOF
-
-if $cc -o $TMPO $TMPC $ffmpeg_cflags $ffmpeg_lflags $LDFLAGS 2> /dev/null  ; then
-    old_ffmpeg_inc="no"
-else
-    old_ffmpeg_inc="yes"
-
-cat > $TMPC << EOF
-#include <ffmpeg/avcodec.h>
-int main(void) {
-    AVFrame *f1, *f2;
-    f2 = av_frame_clone(f1);
-    AV_CODEC_ID_H264;
-    return 0;
-}
-EOF
-
-fi
+old_ffmpeg_inc="no"
 
 cat > $TMPC << EOF
 #include <libavcodec/avcodec.h>
@@ -2807,6 +2788,7 @@
 echo "CXXFLAGS=$CXXFLAGS" >> config.mak
 echo "LDFLAGS=$LDFLAGS" >> config.mak
 echo "SHFLAGS=$SHFLAGS" >> config.mak
+echo "bindir=$bindir" >> config.mak
 echo "libdir=$libdir" >> config.mak
 
 echo "STATIC_MODULES=$static_modules" >> config.mak
