--- gpac-2012-06-06/configure.orig	2012-06-07 15:39:04.581574492 +0000
+++ gpac-2012-06-06/configure	2012-06-07 15:39:44.371832893 +0000
@@ -1209,17 +1209,7 @@
 int main( void ) { return 0; }
 EOF
 
-if $cc -o $TMPO $TMPC $ffmpeg_cflags $ffmpeg_lflags $LDFLAGS 2> /dev/null  ; then
-    old_ffmpeg_inc="no"
-else
-    old_ffmpeg_inc="yes"
-
-    cat > $TMPC << EOF
-#include <ffmpeg/avcodec.h>
-int main( void ) { return 0; }
-EOF
-
-fi
+old_ffmpeg_inc="no"
 
 if test "$cross_prefix" = "" ; then 
     if $cc -o $TMPO $TMPC -I$prefix/include -L$prefix/lib $ffmpeg_lflags $LDFLAGS 2> /dev/null  ; then
