diff -ruN go.orig/src/cmd/compile/internal/gc/noder.go go/src/cmd/compile/internal/gc/noder.go
--- go.orig/src/cmd/compile/internal/gc/noder.go	2018-02-07 17:09:30.000000000 +0000
+++ go/src/cmd/compile/internal/gc/noder.go	2018-02-09 12:42:28.521025700 +0000
@@ -1193,9 +1193,9 @@
 	case strings.HasPrefix(text, "go:cgo_"):
 		// For security, we disallow //go:cgo_* directives outside cgo-generated files.
 		// Exception: they are allowed in the standard library, for runtime and syscall.
-		if !isCgoGeneratedFile(pos) && !compiling_std {
-			p.error(syntax.Error{Pos: pos, Msg: fmt.Sprintf("//%s only allowed in cgo-generated code", text)})
-		}
+		//if !isCgoGeneratedFile(pos) && !compiling_std {
+		//	p.error(syntax.Error{Pos: pos, Msg: fmt.Sprintf("//%s only allowed in cgo-generated code", text)})
+		//}
 		p.pragcgobuf += p.pragcgo(pos, text)
 		fallthrough // because of //go:cgo_unsafe_args
 	default:
