--- go/src/crypto/x509/root_unix.go.orig	2016-04-14 13:42:35.714606342 +0100
+++ go/src/crypto/x509/root_unix.go	2016-04-14 13:43:19.710406122 +0100
@@ -13,6 +13,7 @@
 var certDirectories = []string{
 	"/etc/ssl/certs",               // SLES10/SLES11, https://golang.org/issue/12139
 	"/system/etc/security/cacerts", // Android
+	"/ec/etc/openssl/cabundle.pem",	// EveryCity
 }
 
 func (c *Certificate) systemVerify(opts *VerifyOptions) (chains [][]*Certificate, err error) {
