--- /dev/null	2011-08-05 12:03:18.000000000 +0100
+++ mg-20110120/asprintf.h	2011-08-05 13:37:28.178240265 +0100
@@ -0,0 +1,17 @@
+
+
+#if (defined (__SVR4) && defined (__sun))
+#define bzero(b,n) memset(b,0,n)
+#define bcopy(b1,b2,n) memcpy(b2,b1,n)
+#define bcmp(b1,b2,n) memcmp(b1,b2,n) 
+#ifndef MAX
+# define MAX(a, b) ((a) < (b) ? (b) : (a))
+#endif
+#include <stdio.h>
+#include <stdarg.h>
+#include <stdlib.h>
+
+int vasprintf(char **ret, const char *format, va_list args);
+int asprintf(char **strp, const char *fmt, ...);
+#endif
+
