--- mg-20150323/echo.c.~1~	2015-04-11 13:43:59.127523874 +0000
+++ mg-20150323/echo.c	2015-04-11 13:45:05.370008981 +0000
@@ -10,12 +10,14 @@
  */
 
 #include <sys/queue.h>
+#include <sys/sysmacros.h>
+#include <limits.h>
 #include <signal.h>
 #include <stdarg.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <term.h>
+#include <ncurses/term.h>
 
 #include "def.h"
 #include "funmap.h"
