--- mg-20140414/echo.c.~1~	2014-04-03 20:20:55.000000000 +0100
+++ mg-20140414/echo.c	2014-04-23 20:42:39.227237458 +0100
@@ -15,8 +15,11 @@
 
 #include "funmap.h"
 
+#include <limits.h>
 #include <stdarg.h>
-#include <term.h>
+#include <strings.h>
+#include <sys/sysmacros.h>
+#include </ec/include/ncurses/term.h>
 
 static char	*veread(const char *, char *, size_t, int, va_list);
 static int	 complt(int, int, char *, size_t, int, int *);
