--- /dev/null	2014-04-27 18:55:36.000000000 +0100
+++ mg-20140414/fgetln.c	2014-04-27 19:34:47.035227892 +0100
@@ -0,0 +1,55 @@
+#if defined (__sun) && defined (__SVR4)
+
+#include <stdio.h>
+#include <stdlib.h>
+ 
+#define INITIAL_LINE_LENGTH	256
+
+char *
+fgetln(fp, lenp)
+	register FILE *fp;
+	size_t *lenp;
+{
+	char c;
+	size_t n, siz;
+	size_t len, new_len;
+	char *buf;
+	char *p;
+
+	len = INITIAL_LINE_LENGTH;
+	n = siz = 0;
+
+	if ((buf = malloc(INITIAL_LINE_LENGTH + 1)) == NULL)
+		return (NULL);
+
+	p = buf;
+	for (;;) {
+		if ((c = getc(fp)) == EOF) {
+			if (siz != 0)
+				break;
+			free(buf);
+			return (NULL);
+		}
+
+		++siz;
+ 
+		if (c == '\n') {
+			*p++ = c;
+			break;
+		}
+		if (n++ >= len) {
+			new_len = len << 1;
+			if ((buf = realloc(buf, new_len + 1)) == NULL)
+	                        return (NULL);
+			len = new_len;
+			p = buf;
+	                p += len >> 1;
+		}
+		*p++ = c;
+	}
+	*p = 0;
+	if (lenp != NULL)
+		*lenp = siz;
+	return (buf);
+}
+#endif
