--- mg-20150323/fileio.c.~1~	2015-03-23 11:03:05.000000000 +0000
+++ mg-20150323/fileio.c	2015-04-11 13:59:55.864285773 +0000
@@ -12,7 +12,6 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-#include <dirent.h>
 #include <errno.h>
 #include <fcntl.h>
 #include <limits.h>
@@ -23,6 +22,7 @@
 #include <string.h>
 #include <unistd.h>
 
+#include "adirent.h"
 #include "def.h"
 #include "kbd.h"
 #include "pathnames.h"
@@ -429,7 +429,7 @@
 	size_t		 len, preflen;
 	int		 ret;
 	DIR		*dirp;
-	struct dirent	*dent;
+	struct adirent	*dent;
 	struct list	*last, *current;
 	char		 fl_name[NFILEN + 2];
 	char		 prefixx[NFILEN + 1];
@@ -511,7 +511,7 @@
 		return (NULL);
 	last = NULL;
 
-	while ((dent = readdir(dirp)) != NULL) {
+	while ((dent = READDIR(dirp)) != NULL) {
 		int isdir;
 		if (strncmp(cp, dent->d_name, len) != 0)
 			continue;
