--- mg-20140414/fileio.c.~1~	2014-04-03 20:20:55.000000000 +0100
+++ mg-20140414/fileio.c	2014-04-23 21:27:33.448204202 +0100
@@ -15,11 +15,11 @@
 
 #include <fcntl.h>
 #include <limits.h>
-#include <dirent.h>
 #include <pwd.h>
 #include <string.h>
 #include <unistd.h>
 
+#include "adirent.h"
 #include "kbd.h"
 #include "pathnames.h"
 
@@ -425,7 +424,7 @@
 	size_t		 len, preflen;
 	int		 ret;
 	DIR		*dirp;
-	struct dirent	*dent;
+	struct adirent	*dent;
 	struct list	*last, *current;
 	char		 fl_name[NFILEN + 2];
 	char		 prefixx[NFILEN + 1];
@@ -507,7 +506,7 @@
 		return (NULL);
 	last = NULL;
 
-	while ((dent = readdir(dirp)) != NULL) {
+	while ((dent = READDIR(dirp)) != NULL) {
 		int isdir;
 		if (strncmp(cp, dent->d_name, len) != 0)
 			continue;
