--- mg-20141127/sysdef.h.~1~	2014-11-27 19:26:10.000000000 +0000
+++ mg-20141127/sysdef.h	2014-12-01 09:45:45.164338438 +0000
@@ -6,7 +6,13 @@
  *		POSIX system header file
  */
 #include <sys/types.h>
-#include <sys/queue.h>
+/* This is the queue.h from OpenBSD/NetBSD, works fine. The one provided by
+   others is incompatible. */
+#if !defined(__OPENBSD__) && !defined(__NETBSD__)
+#  include "queue.h"
+#else
+#  include <sys/queue.h>
+#endif
 
 /* necesarry to get asprintf & friends with glibc XXX doesn't work for some
  * mysterious reason! */
@@ -38,3 +44,12 @@
 	mode_t		fi_mode;
 	struct timespec	fi_mtime;	/* Last modified time */
 };
+
+#if defined(__sun) && defined(__SVR4)
+extern char * fgetln(FILE *, size_t *);
+extern void * reallocarray(void *, size_t, size_t);
+extern long long strtonum(const char *, long long, long long, const char **);
+#ifndef LOGIN_NAME_MAX
+#define LOGIN_NAME_MAX LOGNAME_MAX
+#endif
+#endif
