--- mg-20150323/tags.c.~1~	2015-03-23 11:31:48.000000000 +0000
+++ mg-20150323/tags.c	2015-04-11 14:27:44.149237214 +0000
@@ -6,9 +6,7 @@
  * Author: Sunil Nimmagadda <sunil@sunilnimmagadda.com>
  */
 
-#include <sys/queue.h>
 #include <sys/stat.h>
-#include <sys/tree.h>
 #include <sys/types.h>
 #include <ctype.h>
 #include <err.h>
@@ -18,13 +16,19 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#if defined(LIBBSD_OVERLAY) || defined(__FreeBSD__) || defined(__GLIBC__)
-#  include <libutil.h>
+#if defined(__sun) && defined(__SVR4)
+#  include "fparseln.h"
 #else
-#include <util.h>
+#  if defined(LIBBSD_OVERLAY) || defined(__FreeBSD__) || defined(__GLIBC__)
+#    include <libutil.h>
+#  else
+#    include <util.h>
+#  endif
 #endif
 
+#include "queue.h"
 #include "def.h"
+#include "tree.h"
 
 struct ctag;
 
