--- mg-20140414/ttyio.c.~1~	2014-04-03 20:20:56.000000000 +0100
+++ mg-20140414/ttyio.c	2014-04-24 00:53:41.789513872 +0100
@@ -11,13 +11,14 @@
  */
 #include "def.h"
 
+#include <sys/filio.h>
+#include <sys/ioctl.h>
 #include <sys/types.h>
 #include <sys/time.h>
-#include <sys/ioctl.h>
 #include <fcntl.h>
 #include <poll.h>
 #include <termios.h>
-#include <term.h>
+#include </ec/include/ncurses/term.h>
 
 #define NOBUF	512			/* Output buffer size. */
 
