--- mg-20150323/ttyio.c.~1~	2015-03-23 11:03:05.000000000 +0000
+++ mg-20150323/ttyio.c	2015-04-11 14:10:00.949226283 +0000
@@ -10,6 +10,7 @@
  * buffered fashion.
  */
 
+#include <sys/filio.h>
 #include <sys/ioctl.h>
 #include <sys/queue.h>
 #include <sys/time.h>
@@ -21,7 +22,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <term.h>
+#include <ncurses/term.h>
 #include <termios.h>
 #include <unistd.h>
 
