--- libvpx-v1.2.0/build/make/Makefile.orig	2013-12-11 23:20:45.314190057 +0000
+++ libvpx-v1.2.0/build/make/Makefile	2013-12-11 23:22:18.688797541 +0000
@@ -211,7 +211,7 @@
 $(1): $(filter-out -%,$(2))
 $(1):
 	$(if $(quiet),@echo    "    [LD] $$@")
-	$(qexec)$$(LD) $$(strip $$(INTERNAL_LDFLAGS) $$(LDFLAGS) -o $$@ $(2) $(3) $$(extralibs))
+	$$(CC) $$(INTERNAL_LDFLAGS) $$(LDFLAGS) -o $$@ $(2) $(3) $$(extralibs)
 endef
 define linkerxx_template
 $(1): $(filter-out -%,$(2))
