--- libxslt-1.1.33/configure.ac.orig	2019-03-20 13:15:17.061674950 +0000
+++ libxslt-1.1.33/configure.ac	2019-03-20 13:16:24.449174499 +0000
@@ -300,6 +300,10 @@
     else
         pythondir='$(libdir)/python$(PYTHON_VERSION)/site-packages'
     fi
+    if test "$PYTHON_VENDOR_PACKAGES" != ""
+    then
+        PYTHON_SITE_PACKAGES=$PYTHON_VENDOR_PACKAGES
+    fi
 fi
 AM_CONDITIONAL(WITH_PYTHON, test "$PYTHON_INCLUDES" != "")
 if test "$PYTHON_INCLUDES" != ""
