diff -ruN logrotate-3.8.1.orig/Makefile logrotate-3.8.1/Makefile
--- logrotate-3.8.1.orig/Makefile	2011-08-31 13:39:04.000000000 +0100
+++ logrotate-3.8.1/Makefile	2011-09-12 16:56:05.368375414 +0100
@@ -80,7 +80,7 @@
     CFLAGS += -DSTATEFILE=\"$(STATEFILE)\"
 endif
 
-BINDIR = $(BASEDIR)/sbin
+BINDIR = $(BASEDIR)/bin
 MANDIR ?= $(BASEDIR)/man
 
 #--------------------------------------------------------------------------
