--- logrotate-3.8.1/config.h.orig	2012-12-20 14:22:16.983686521 +0000
+++ logrotate-3.8.1/config.h	2012-12-20 14:23:44.756525848 +0000
@@ -13,8 +13,8 @@
 
 #ifdef SunOS
 #define DEFAULT_MAIL_COMMAND "/usr/bin/mailx"
-#define COMPRESS_COMMAND "/usr/local/bin/gzip"
-#define UNCOMPRESS_COMMAND "/usr/local/bin/gunzip"
+#define COMPRESS_COMMAND "/ec/bin/gzip"
+#define UNCOMPRESS_COMMAND "/ec/bin/gunzip"
 #define STATEFILE "/var/log/logrotate.status"
 #endif
 
