--- dwarf-20150310/libdwarf/Makefile.in.~1~	2015-03-03 15:40:11.000000000 +0000
+++ dwarf-20150310/libdwarf/Makefile.in	2015-04-02 13:07:30.395570131 +0000
@@ -122,7 +122,7 @@
 	$(AR) $(ARFLAGS) $@ $(OBJS)  dwarf_names.o > ar-output-temp
 
 libdwarf.so: dwarf_names.h dwarf_names.c $(OBJS) dwarf_names.o
-	$(CC) $(CFLAGS) -shared $(OBJS)  dwarf_names.o -o $@
+	$(CC) $(CFLAGS) -shared $(OBJS)  dwarf_names.o -o $@ $(LDFLAGS) $(LIBS)
 
 none:
 	echo "do nothing " $@
