--- phantomjs-1.9.8/src/phantomjs.pro.orig	2014-11-17 13:44:42.730638057 +0000
+++ phantomjs-1.9.8/src/phantomjs.pro	2014-11-17 13:45:38.996314655 +0000
@@ -131,3 +131,21 @@
             qico
     }
 }
+
+solaris* {
+    LIBS += -lnsl -lsocket
+
+    QTPLUGIN += \
+    qcncodecs \
+    qjpcodecs \
+    qkrcodecs \
+    qtwcodecs
+}
+
+netbsd* {
+    QTPLUGIN += \
+    qcncodecs \
+    qjpcodecs \
+    qkrcodecs \
+    qtwcodecs
+}
