diff -ruN icinga2-2.4.1.orig/lib/base/math-script.cpp icinga2-2.4.1/lib/base/math-script.cpp
--- icinga2-2.4.1.orig/lib/base/math-script.cpp	2015-11-26 10:19:33.000000000 +0000
+++ icinga2-2.4.1/lib/base/math-script.cpp	2015-12-08 16:03:38.691847247 +0000
@@ -27,6 +27,8 @@
 #include <boost/foreach.hpp>
 #include <cmath>
 
+#define INFINITY (__builtin_huge_val ())
+
 using namespace icinga;
 
 static double MathAbs(double x)
