--- varnish-3.0.3/bin/varnishd/cache_waiter_ports.c.orig	2012-10-10 09:22:01.179850444 +0000
+++ varnish-3.0.3/bin/varnishd/cache_waiter_ports.c	2012-10-10 09:22:39.738955687 +0000
@@ -225,7 +225,7 @@
 
 		if (sp) {
 			double tmo =
-			    (sp->t_end + cache_param->timeout_idle) - now;
+			    (sp->t_end + params->sess_timeout) - now;
 
 			/* we should have removed all sps whose timeout has passed */
 			assert(tmo > 0.0);
