--- varnish-4.0.2/configure.orig	2014-12-04 12:09:12.265963812 +0000
+++ varnish-4.0.2/configure	2014-12-04 12:09:19.591157322 +0000
@@ -16794,7 +16794,7 @@
 # The reason for -Wno-error=unused-result is a glibc/gcc interaction
 # idiocy where write is marked as warn_unused_result, causing build
 # failures.
-CFLAGS="${CFLAGS} -Wall -Werror"
+CFLAGS="${CFLAGS} -Wall"
 OCFLAGS="${OCFLAGS} -Wall -Werror"
 as_CACHEVAR=`$as_echo "ax_cv_check_cflags__-Werror=unused-result" | $as_tr_sh`
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking whether C compiler accepts -Werror=unused-result" >&5
