diff --git a/doc/sphinx/Makefile.am b/doc/sphinx/Makefile.am
index ba098ca..c82dec8 100644
--- a/doc/sphinx/Makefile.am
+++ b/doc/sphinx/Makefile.am
@@ -148,12 +148,12 @@ reference:
 	test -d $@ || mkdir $@
 BUILT_SOURCES += reference
 
-reference/vmod_std.generated.rst: $(top_builddir)/lib/libvmod_std/vmod_std.rst
-	cp $? $@
+reference/vmod_std.generated.rst: reference $(top_builddir)/lib/libvmod_std/vmod_std.rst
+	cp $(top_builddir)/lib/libvmod_std/vmod_std.rst $@
 BUILT_SOURCES += reference/vmod_std.generated.rst
 
-reference/vmod_directors.generated.rst: $(top_builddir)/lib/libvmod_directors/vmod_directors.rst
-	cp $? $@
+reference/vmod_directors.generated.rst: reference $(top_builddir)/lib/libvmod_directors/vmod_directors.rst
+	cp $(top_builddir)/lib/libvmod_directors/vmod_directors.rst $@
 BUILT_SOURCES += reference/vmod_directors.generated.rst
 
 EXTRA_DIST += $(BUILT_SOURCES)
