diff -ruN mongodb-src-r2.0.4.orig/SConstruct mongodb-src-r2.0.4/SConstruct
--- mongodb-src-r2.0.4.orig/SConstruct	2012-03-19 16:10:40.000000000 +0000
+++ mongodb-src-r2.0.4/SConstruct	2012-05-01 16:46:37.117629002 +0100
@@ -1453,9 +1453,12 @@
         e.Alias( name , name + ".exe" )
         name += ".exe"
 
-    inst = e.Install( installDir + "/bin" , name )
-
-    fullInstallName = installDir + "/bin/" + name
+    if force64:
+	inst = e.Install( installDir + "/bin/amd64" , name )
+	fullInstallName = installDir + "/bin/amd64/" + name
+    else:
+	inst = e.Install( installDir + "/bin" , name )
+	fullInstallName = installDir + "/bin/" + name
 
     allBinaries += [ name ]
     if (solaris or linux) and (not has_option("nostrip")):
