--- pycurl-7.43.0/src/easy.c.orig	2016-02-04 04:09:31.000000000 +0000
+++ pycurl-7.43.0/src/easy.c	2017-01-23 14:25:26.699298848 +0000
@@ -653,12 +653,12 @@
 #if !defined(WIN32)
     case AF_UNIX:
         {
-            struct sockaddr_un* sun = (struct sockaddr_un*)saddr;
+            struct sockaddr_un* s_un = (struct sockaddr_un*)saddr;
 
 #if PY_MAJOR_VERSION >= 3
-            res_obj = Py_BuildValue("y", sun->sun_path);
+            res_obj = Py_BuildValue("y", s_un->sun_path);
 #else
-            res_obj = Py_BuildValue("s", sun->sun_path);
+            res_obj = Py_BuildValue("s", s_un->sun_path);
 #endif
         }
         break;
