From c94127d882fc25a40bd8505232b6cbaa99b25d8c Mon Sep 17 00:00:00 2001
From: Eric Stanley <estanley@nagios.com>
Date: Mon, 18 Aug 2014 06:29:27 -0500
Subject: [PATCH] JSON CGI: Fixed memory overwrite issue in string escaping.

---
 cgi/jsonutils.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/cgi/jsonutils.c b/cgi/jsonutils.c
index 2256652..59c0ac5 100644
--- a/cgi/jsonutils.c
+++ b/cgi/jsonutils.c
@@ -1390,7 +1390,7 @@ char *json_escape_string(const char *src, const json_escape *escapes) {
 
 	/* Make a wide string copy of src */
 	wdest_len = mbstowcs(NULL, src, 0);
-	if(0 == wdest_len) return NULL;
+	if(wdest_len <= 0) return NULL;
 	if((wdest = calloc(wdest_len + 1, sizeof(wchar_t))) == NULL) {
 		return NULL;
 		}
@@ -1412,11 +1412,11 @@ char *json_escape_string(const char *src, const json_escape *escapes) {
 				if((wdest_size - wdest_len) < (to_len - from_len)) {
 					/* If more room is needed, realloc and update variables */
 					wdest_size += (to_len - from_len) * BUF_REALLOC_MULTIPLIER;
-					wdest = realloc(wdest, wdest_size * sizeof(wchar_t));
+					wdest = realloc(wdest, (wdest_size + 1) * sizeof(wchar_t));
 					if(NULL == wdest) return NULL;
 					fromp = wdest + offset;
 					}
-				wchars = wdest_len - offset + from_len + 1;
+				wchars = wdest_len - offset - from_len + 1;
 				wmemmove(fromp + to_len, fromp + from_len, wchars);
 				wcsncpy(fromp, escp->to, to_len);
 				wdest_len += (to_len - from_len);
