--- subversion-1.7.7/build/get-py-info.py.~1~	2009-11-16 19:07:17.000000000 +0000
+++ subversion-1.7.7/build/get-py-info.py	2012-12-20 16:07:01.081045500 +0000
@@ -28,6 +28,7 @@
 
 import sys
 import os
+import re
 
 def usage():
   print('USAGE: python %s WHAT' % sys.argv[0])
@@ -61,6 +62,7 @@
 
 if sys.argv[1] == '--compile':
   cc, ccshared = sysconfig.get_config_vars('CC', 'CCSHARED')
+  cc = cc.replace('-m32', '').replace('  ',' ')
   print("%s %s" % (cc, ccshared))
   sys.exit(0)
 
@@ -121,6 +123,7 @@
     python_version = sysconfig.get_config_var('VERSION')
     add_option_if_missing(options, "-lpython%s" % python_version)
 
+  options = [option for option in options if option != '-m32']
   return options
 
 def lib_options():
