Index: curl-7.36.0/include/curl/Makefile.in
===================================================================
--- curl-7.36.0/include/curl/Makefile.in.orig
+++ curl-7.36.0/include/curl/Makefile.in
@@ -436,11 +436,7 @@ $(ACLOCAL_M4): @MAINTAINER_MODE_TRUE@ $(
 	cd $(top_builddir) && $(MAKE) $(AM_MAKEFLAGS) am--refresh
 $(am__aclocal_m4_deps):
 
-curlbuild.h: stamp-h2
-	@test -f $@ || rm -f stamp-h2
-	@test -f $@ || $(MAKE) $(AM_MAKEFLAGS) stamp-h2
-
-stamp-h2: $(srcdir)/curlbuild.h.in $(top_builddir)/config.status
+stamp-h2:
 	@rm -f stamp-h2
 	cd $(top_builddir) && $(SHELL) ./config.status include/curl/curlbuild.h
 
