diff -ruN perl-5.20.0.orig/hints/solaris_2.sh perl-5.20.0/hints/solaris_2.sh
--- perl-5.20.0.orig/hints/solaris_2.sh	2014-05-26 13:34:20.000000000 +0000
+++ perl-5.20.0/hints/solaris_2.sh	2014-06-18 13:27:17.906361775 +0000
@@ -308,8 +308,8 @@
 doesn't work, you should use -B/usr/ccs/bin/ instead.
 
 END
-		ccdlflags="$ccdlflags -Wl,-E"
-		lddlflags="$lddlflags -Wl,-E -shared"
+		ccdlflags="$ccdlflags"
+		lddlflags="$lddlflags -shared"
 	    fi
 	fi
 
@@ -623,7 +623,7 @@
 		esac
 		ccflags="$ccflags $getconfccflags"
 		ldflags="$ldflags $getconfldflags"
-		lddlflags="$lddlflags -G $getconflddlflags"
+		lddlflags="$lddlflags -shared $getconflddlflags"
 
 		echo "int main() { return(0); } " > try.c
 		tryworkshopcc="${cc:-cc} try.c -o try $ccflags"
