Index: openssh-6.0p1/session.c
===================================================================
--- openssh-6.0p1/session.c.orig
+++ openssh-6.0p1/session.c
@@ -1200,9 +1200,25 @@ do_setup_env(Session *s, const char *she
 		/* Normal systems set SHELL by default. */
 		child_set_env(&env, &envsize, "SHELL", shell);
 	}
+
+#define PASS_ENV(x) \
+	if (getenv(x)) \
+		child_set_env(&env, &envsize, x, getenv(x));
+
 	if (getenv("TZ"))
 		child_set_env(&env, &envsize, "TZ", getenv("TZ"));
 
+	PASS_ENV("LANG")
+	PASS_ENV("LC_ALL")
+	PASS_ENV("LC_CTYPE")
+	PASS_ENV("LC_COLLATE")
+	PASS_ENV("LC_TIME")
+	PASS_ENV("LC_NUMERIC")
+	PASS_ENV("LC_MONETARY")
+	PASS_ENV("LC_MESSAGES")
+	/*PASS_ENV("LINES")*/
+	/*PASS_ENV("COLUMNS")*/
+
 	/* Set custom environment options from RSA authentication. */
 	if (!options.use_login) {
 		while (custom_environment) {
