--- openssh-6.4p1/sshd_config.orig	2013-11-10 19:05:11.961287809 +0000
+++ openssh-6.4p1/sshd_config	2013-11-10 19:06:01.947468946 +0000
@@ -1,10 +1,6 @@
-#	$OpenBSD: sshd_config,v 1.90 2013/05/16 04:09:14 dtucker Exp $
-
 # This is the sshd server system-wide configuration file.  See
 # sshd_config(5) for more information.
 
-# This sshd was compiled with PATH=/usr/bin:/bin:/usr/sbin:/sbin
-
 # The strategy used for options in the default sshd_config shipped with
 # OpenSSH is to specify options with their default value where
 # possible, but leave them commented.  Uncommented options override the
@@ -12,7 +8,7 @@
 
 #Port 22
 #AddressFamily any
-#ListenAddress 0.0.0.0
+ListenAddress 0.0.0.0
 #ListenAddress ::
 
 # The default requires explicit activation of protocol 1
@@ -110,7 +106,7 @@
 #Compression delayed
 #ClientAliveInterval 0
 #ClientAliveCountMax 3
-#UseDNS yes
+UseDNS no
 #PidFile /var/run/sshd.pid
 #MaxStartups 10:30:100
 #PermitTunnel no
