--- openssh-6.5p1/sshd_config.orig	2014-01-31 16:36:24.133613309 +0000
+++ openssh-6.5p1/sshd_config	2014-01-31 16:37:44.063297381 +0000
@@ -3,8 +3,6 @@
 # This is the sshd server system-wide configuration file.  See
 # sshd_config(5) for more information.
 
-# This sshd was compiled with PATH=/usr/bin:/bin:/usr/sbin:/sbin
-
 # The strategy used for options in the default sshd_config shipped with
 # OpenSSH is to specify options with their default value where
 # possible, but leave them commented.  Uncommented options override the
@@ -12,7 +10,7 @@
 
 #Port 22
 #AddressFamily any
-#ListenAddress 0.0.0.0
+ListenAddress 0.0.0.0
 #ListenAddress ::
 
 # The default requires explicit activation of protocol 1
