--- readline-6.2/shlib/Makefile.in.orig	2011-10-13 14:27:53.249383525 +0100
+++ readline-6.2/shlib/Makefile.in	2011-10-13 14:28:23.087235674 +0100
@@ -174,7 +174,7 @@
 
 $(SHARED_READLINE):	$(SHARED_OBJ)
 	$(RM) $@
-	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS)
+	$(SHOBJ_LD) ${SHOBJ_LDFLAGS} ${SHLIB_XLDFLAGS} -o $@ $(SHARED_OBJ) $(SHLIB_LIBS) ${TERMCAP_LIB}
 
 $(SHARED_HISTORY):	$(SHARED_HISTOBJ) xmalloc.so xfree.so
 	$(RM) $@
