From 028ff36f20ffd5d5edf79af662e34908ce031a36 Mon Sep 17 00:00:00 2001
From: Maciej Blizinski <maciej@opencsw.org>
Date: Thu, 14 Jun 2012 12:12:06 +0200
Subject: [PATCH] Add -h to the linker invocation

---
 Makefile.in |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index 241deed..13bb227 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -160,7 +160,7 @@ minigzip64.o: test/minigzip.c zlib.h zconf.h
 	-@mv objs/$*.o $@
 
 placebo $(SHAREDLIBV): $(PIC_OBJS) libz.a
-	$(LDSHARED) $(SFLAGS) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS)
+	$(LDSHARED) $(SFLAGS) -h $(SHAREDLIBM) -o $@ $(PIC_OBJS) $(LDSHAREDLIBC) $(LDFLAGS)
 	rm -f $(SHAREDLIB) $(SHAREDLIBM)
 	ln -s $@ $(SHAREDLIB)
 	ln -s $@ $(SHAREDLIBM)
-- 
1.7.10.3

