--- zlib-1.2.7/configure.~1~	2012-05-02 04:17:08.000000000 +0000
+++ zlib-1.2.7/configure	2012-12-26 23:40:14.024693860 +0000
@@ -64,7 +64,6 @@
 libdir=${libdir-'${exec_prefix}/lib'}
 sharedlibdir=${sharedlibdir-'${libdir}'}
 includedir=${includedir-'${prefix}/include'}
-mandir=${mandir-'${prefix}/share/man'}
 shared_ext='.so'
 shared=1
 solo=0
@@ -110,6 +109,8 @@
     esac
 done
 
+mandir=${prefix}/share/man
+
 # define functions for testing compiler and library characteristics and logging the results
 test=ztest$$
 
