diff -ruN cronie-1.4.8.orig/src/Makefile.am cronie-1.4.8/src/Makefile.am
--- cronie-1.4.8.orig/src/Makefile.am	2010-12-16 08:59:02.000000000 +0000
+++ cronie-1.4.8/src/Makefile.am	2011-07-20 14:43:53.914877642 +0100
@@ -43,7 +43,7 @@
 			 * didn't exist last time around...) \
 			 * or it will be checked by inotify \
 			 */ \
-	#define SPOOL_DIR	"$(SPOOL_DIR)" \
+	#define SPOOL_DIR	"$(SPOOL_DIR)/crontabs" \
 	\
 			/* CRON_HOSTNAME is file in SPOOL_DIR which, if it \
 			 * exists, and does not just contain a line matching \
@@ -58,8 +58,8 @@
 			/* cron allow/deny file.  At least cron.deny must \
 			 * exist for ordinary users to run crontab. \
 			 */ \
-	#define	CRON_ALLOW	"$(sysconfdir)/cron.allow" \
-	#define	CRON_DENY	"$(sysconfdir)/cron.deny" \
+	#define	CRON_ALLOW	"$(SYS_CROND_DIR)/cron.allow" \
+	#define	CRON_DENY	"$(SYS_CROND_DIR)/cron.deny" \
 	\
 			/* 4.3BSD-style crontab f.e. /etc/crontab */ \
 	#define SYSCRONTAB	"$(SYSCRONTAB)" \
