--- cronie-1.5.1/contrib/0anacron.orig	2016-12-13 15:04:22.098998856 +0000
+++ cronie-1.5.1/contrib/0anacron	2016-12-13 15:05:59.362603027 +0000
@@ -8,10 +8,10 @@
 fi
 
 # Do not run jobs when on battery power
-if test -x /usr/bin/on_ac_power; then
-    /usr/bin/on_ac_power >/dev/null 2>&1
-    if test $? -eq 1; then
-    exit 0
-    fi
-fi
-/usr/sbin/anacron -s
+#if test -x /usr/bin/on_ac_power; then
+#    /usr/bin/on_ac_power >/dev/null 2>&1
+#    if test $? -eq 1; then
+#    exit 0
+#    fi
+#fi
+/ec/bin/anacron -s
diff -ruN cronie-1.4.8.orig/contrib/0hourly cronie-1.4.8/contrib/0hourly
--- cronie-1.4.8.orig/contrib/0hourly	2010-12-16 08:59:02.000000000 +0000
+++ cronie-1.4.8/contrib/0hourly	2011-07-20 15:03:25.999379581 +0100
@@ -2,4 +2,4 @@
 SHELL=/bin/bash
-PATH=/sbin:/bin:/usr/sbin:/usr/bin
+PATH=/ec/bin:/sbin:/bin:/usr/sbin:/usr/bin
 MAILTO=root
-01 * * * * root run-parts /etc/cron.hourly
+01 * * * * root /ec/bin/run-parts /etc/cron.hourly
diff -ruN cronie-1.4.8.orig/contrib/anacrontab cronie-1.4.8/contrib/anacrontab
--- cronie-1.4.8.orig/contrib/anacrontab	2010-12-16 08:59:02.000000000 +0000
+++ cronie-1.4.8/contrib/anacrontab	2011-07-20 15:03:25.999596731 +0100
@@ -2,8 +2,8 @@
 
 # See anacron(8) and anacrontab(5) for details.
 
-SHELL=/bin/sh
-PATH=/sbin:/bin:/usr/sbin:/usr/bin
+SHELL=/bin/bash
+PATH=/ec/bin:/sbin:/bin:/usr/sbin:/usr/bin
 MAILTO=root
 # the maximal random delay added to the base delay of the jobs
 RANDOM_DELAY=45
@@ -11,6 +11,6 @@
 START_HOURS_RANGE=3-22
 
 #period in days   delay in minutes   job-identifier   command
-1	5	cron.daily		nice run-parts /etc/cron.daily
-7	25	cron.weekly		nice run-parts /etc/cron.weekly
-@monthly 45	cron.monthly		nice run-parts /etc/cron.monthly
+1	5	cron.daily		nice /ec/bin/run-parts /etc/cron.daily
+7	25	cron.weekly		nice /ec/bin/run-parts /etc/cron.weekly
+@monthly 45	cron.monthly		nice /ec/bin/run-parts /etc/cron.monthly
