diff -ruN cronie-1.5.1.orig/contrib/0anacron cronie-1.5.1/contrib/0anacron
--- cronie-1.5.1.orig/contrib/0anacron	2013-01-17 13:04:09.000000000 +0000
+++ cronie-1.5.1/contrib/0anacron	2017-06-14 15:04:37.009005848 +0000
@@ -8,10 +8,10 @@
 fi
 
 # Do not run jobs when on battery power
-if test -x /usr/bin/on_ac_power; then
-    /usr/bin/on_ac_power >/dev/null 2>&1
-    if test $? -eq 1; then
-    exit 0
-    fi
-fi
+#if test -x /usr/bin/on_ac_power; then
+#    /usr/bin/on_ac_power >/dev/null 2>&1
+#    if test $? -eq 1; then
+#    exit 0
+#    fi
+#fi
 /usr/sbin/anacron -s
diff -ruN cronie-1.5.1.orig/contrib/0hourly cronie-1.5.1/contrib/0hourly
--- cronie-1.5.1.orig/contrib/0hourly	2011-12-06 17:50:01.000000000 +0000
+++ cronie-1.5.1/contrib/0hourly	2017-06-14 15:03:57.311808169 +0000
@@ -1,5 +1,5 @@
 # Run the hourly jobs
 SHELL=/bin/bash
-PATH=/sbin:/bin:/usr/sbin:/usr/bin
+PATH=/ec/bin:/sbin:/bin:/usr/sbin:/usr/bin
 MAILTO=root
-01 * * * * root run-parts /etc/cron.hourly
+01 * * * * root /ec/bin/run-parts /etc/cron.hourly
diff -ruN cronie-1.5.1.orig/contrib/0hourly.orig cronie-1.5.1/contrib/0hourly.orig
--- cronie-1.5.1.orig/contrib/0hourly.orig	1970-01-01 00:00:00.000000000 +0000
+++ cronie-1.5.1/contrib/0hourly.orig	2011-12-06 17:50:01.000000000 +0000
@@ -0,0 +1,5 @@
+# Run the hourly jobs
+SHELL=/bin/bash
+PATH=/sbin:/bin:/usr/sbin:/usr/bin
+MAILTO=root
+01 * * * * root run-parts /etc/cron.hourly
diff -ruN cronie-1.5.1.orig/contrib/anacrontab cronie-1.5.1/contrib/anacrontab
--- cronie-1.5.1.orig/contrib/anacrontab	2009-08-28 10:55:51.000000000 +0000
+++ cronie-1.5.1/contrib/anacrontab	2017-06-14 15:03:57.312020296 +0000
@@ -2,8 +2,8 @@
 
 # See anacron(8) and anacrontab(5) for details.
 
-SHELL=/bin/sh
-PATH=/sbin:/bin:/usr/sbin:/usr/bin
+SHELL=/bin/bash
+PATH=/ec/bin:/sbin:/bin:/usr/sbin:/usr/bin
 MAILTO=root
 # the maximal random delay added to the base delay of the jobs
 RANDOM_DELAY=45
@@ -11,6 +11,6 @@
 START_HOURS_RANGE=3-22
 
 #period in days   delay in minutes   job-identifier   command
-1	5	cron.daily		nice run-parts /etc/cron.daily
-7	25	cron.weekly		nice run-parts /etc/cron.weekly
-@monthly 45	cron.monthly		nice run-parts /etc/cron.monthly
+1	5	cron.daily		nice /ec/bin/run-parts /etc/cron.daily
+7	25	cron.weekly		nice /ec/bin/run-parts /etc/cron.weekly
+@monthly 45	cron.monthly		nice /ec/bin/run-parts /etc/cron.monthly
