--- Net-SSLeay-1.55/inc/Module/Install/PRIVATE/Net/SSLeay.pm.backup	2013-12-13 19:01:54.401441861 +0000
+++ Net-SSLeay-1.55/inc/Module/Install/PRIVATE/Net/SSLeay.pm	2013-12-13 19:02:13.906970319 +0000
@@ -176,19 +176,7 @@
     }
 
     my @guesses = (
-            '/usr/bin/openssl'               => '/usr',
-            '/usr/sbin/openssl'              => '/usr',
-            '/opt/ssl/bin/openssl'           => '/opt/ssl',
-            '/opt/ssl/sbin/openssl'          => '/opt/ssl',
-            '/usr/local/ssl/bin/openssl'     => '/usr/local/ssl',
-            '/usr/local/openssl/bin/openssl' => '/usr/local/openssl',
-            '/apps/openssl/std/bin/openssl'  => '/apps/openssl/std',
-            '/usr/sfw/bin/openssl'           => '/usr/sfw', # Open Solaris
-            'C:\OpenSSL\bin\openssl.exe'     => 'C:\OpenSSL',
-            $Config{prefix} . '\bin\openssl.exe'      => $Config{prefix},           # strawberry perl
-            $Config{prefix} . '\..\c\bin\openssl.exe' => $Config{prefix} . '\..\c', # strawberry perl
-            '/sslexe/openssl.exe'            => '/sslroot',  # VMS, openssl.org
-            '/ssl$exe/openssl.exe'           => '/ssl$root', # VMS, HP install
+            '/ec/bin/openssl'               => '/ec',
     );
 
     while (my $k = shift @guesses
