--- Net-SSLeay-1.55/inc/Module/Install/PRIVATE/Net/SSLeay.pm.~1~	2013-05-21 23:43:07.000000000 +0100
+++ Net-SSLeay-1.55/inc/Module/Install/PRIVATE/Net/SSLeay.pm	2013-12-14 13:28:17.404977962 +0000
@@ -47,15 +47,15 @@
         LIBS => join(' ', (map {"-L$_"} @{$opts->{lib_paths}}), (map {"-l$_"} @{$opts->{lib_links}})),
     );
 
-    if ( $self->prompt(
-            "Do you want to run external tests?\n".
-            "These tests *will* *fail* if you do not have network connectivity.",
-            'n',
-    ) =~ /^y/i ) {
+#    if ( $self->prompt(
+#            "Do you want to run external tests?\n".
+#            "These tests *will* *fail* if you do not have network connectivity.",
+#            'n',
+#    ) =~ /^y/i ) {
         $self->tests('t/*/*.t t/*/*/*.t');
-    } else {
-        $self->tests('t/local/*.t t/handle/local/*.t');
-    }
+#    } else {
+#        $self->tests('t/local/*.t t/handle/local/*.t');
+#    }
 }
 
 sub ssleay_get_build_opts {
@@ -176,19 +176,7 @@
     }
 
     my @guesses = (
-            '/usr/bin/openssl'               => '/usr',
-            '/usr/sbin/openssl'              => '/usr',
-            '/opt/ssl/bin/openssl'           => '/opt/ssl',
-            '/opt/ssl/sbin/openssl'          => '/opt/ssl',
-            '/usr/local/ssl/bin/openssl'     => '/usr/local/ssl',
-            '/usr/local/openssl/bin/openssl' => '/usr/local/openssl',
-            '/apps/openssl/std/bin/openssl'  => '/apps/openssl/std',
-            '/usr/sfw/bin/openssl'           => '/usr/sfw', # Open Solaris
-            'C:\OpenSSL\bin\openssl.exe'     => 'C:\OpenSSL',
-            $Config{prefix} . '\bin\openssl.exe'      => $Config{prefix},           # strawberry perl
-            $Config{prefix} . '\..\c\bin\openssl.exe' => $Config{prefix} . '\..\c', # strawberry perl
-            '/sslexe/openssl.exe'            => '/sslroot',  # VMS, openssl.org
-            '/ssl$exe/openssl.exe'           => '/ssl$root', # VMS, HP install
+            '/ec/bin/openssl'               => '/ec',
     );
 
     while (my $k = shift @guesses
