--- FreeImage.orig/Makefile.solaris	2009-07-30 06:35:48.000000000 +0100
+++ FreeImage/Makefile.solaris	2011-04-05 18:21:55.023851054 +0100
@@ -5,27 +5,25 @@
 include Makefile.srcs
 
 # General configuration variables:
-CC = gcc
-CPP = g++
+CPP = $(CXX)
 AR = ar
 
-INSTALLDIR = /usr/local
+INSTALLDIR = $(DESTDIR)
 
 # Converts cr/lf to just lf
 DOS2UNIX = dos2unix
 
-COMPILERFLAGS = -O3
-LIBRARIES = -lstdc++
+COMPILERFLAGS = 
+LIBRARIES = 
 
 MODULES = $(SRCS:.c=.o)
 MODULES := $(MODULES:.cpp=.o)
-CFLAGS = $(COMPILERFLAGS) $(INCLUDE)
-CPPFLAGS = $(COMPILERFLAGS)  -Wno-ctor-dtor-privacy $(INCLUDE)
+CPPFLAGS = $(INCLUDE)
 
 TARGET  = freeimage
-STATICLIB = lib$(TARGET).a
 SHAREDLIB = lib$(TARGET)-$(VER_MAJOR).$(VER_MINOR).so
 LIBNAME = lib$(TARGET).so.$(VER_MAJOR)
+SHORTLIBNAME = lib$(TARGET).so
 
 
 
@@ -34,32 +32,33 @@ default: all
 all: dist
 
 dist: FreeImage
-	cp *.a Dist
 	cp *.so Dist
 	cp Source/FreeImage.h Dist
 
 dos2unix:
 	@$(DOS2UNIX) $(SRCS) $(INCLS)
 
-FreeImage: $(STATICLIB) $(SHAREDLIB)
+FreeImage: $(SHAREDLIB)
 
 .c.o:
-	$(CC) $(CFLAGS) -c $< -o $@
+	$(CC) $(CPPFLAGS) $(CFLAGS) -c $< -o $@
 
 .cpp.o:
-	$(CPP) $(CPPFLAGS) -c $< -o $@
+	$(CPP) $(CPPFLAGS) $(CXXFLAGS) -c $< -o $@
 
 $(STATICLIB): $(MODULES)
 	$(AR) r $@ $(MODULES)
 
 $(SHAREDLIB): $(MODULES)
-	$(CC) -s -G -Wl,-soname=$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
+	$(CC) -s -G -Wl,-h$(LIBNAME) -o $@ $(MODULES) $(LIBRARIES)
 
 install:
-	install -m 644 -o root -g root Source/FreeImage.h $(INSTALLDIR)/include
-	install -m 644 -o root -g root $(STATICLIB) $(INSTALLDIR)/lib
-	install -m 755 -o root -g root $(SHAREDLIB) $(INSTALLDIR)/lib
-	ln -sf $(SHAREDLIB) $(INSTALLDIR)/lib/$(LIBNAME)
+	mkdir -p $(INSTALLDIR)/include
+	mkdir -p $(INSTALLLIBDIR)
+	install -m 644 Source/FreeImage.h $(INSTALLDIR)/include
+	install -m 755 $(SHAREDLIB) $(INSTALLLIBDIR)/
+	ln -sf $(SHAREDLIB) $(INSTALLLIBDIR)/$(LIBNAME)
+	ln -sf $(SHAREDLIB) $(INSTALLLIBDIR)/$(SHORTLIBNAME)
 
 clean:
-	rm -rf core Dist/*.* u2dtmp* $(MODULES) $(STATICLIB) $(SHAREDLIB) $(LIBNAME)
+	rm -rf core Dist/*.* u2dtmp* $(MODULES) $(SHAREDLIB) $(LIBNAME)
