diff -ruN mailwrapper-opencsw-1.0.orig/Makefile mailwrapper-opencsw-1.0/Makefile
--- mailwrapper-opencsw-1.0.orig/Makefile	2012-06-19 14:29:22.000000000 +0000
+++ mailwrapper-opencsw-1.0/Makefile	2012-08-17 13:04:30.928949784 +0000
@@ -24,9 +24,9 @@
 	-rm $(LDOBJS)
 
 install:
-	-ginstall -D -m 644 mailer.conf $(DESTDIR)/$(sysconfdir)/mailer.conf
-	-ginstall -D -m 755 mailwrapper $(DESTDIR)/$(sbindir)/mailwrapper
-	-ginstall -D -m 755 mailwrapper.8 $(DESTDIR)/$(mandir)/man8/mailwrapper.8
+	-install -D -m 644 mailer.conf $(DESTDIR)/$(ECPREFIX)/etc/mailer.conf
+	-install -D -m 755 mailwrapper $(DESTDIR)/$(ECPREFIX)/bin/mailwrapper
+	-install -D -m 755 mailwrapper.8 $(DESTDIR)/$(ECPREFIX)/share/man8/mailwrapper.8
 
 mailwrapper: $(INCS) $(LDOBJS) Makefile
 	$(CC) $(LDFLAGS) -o mailwrapper $(LDOBJS)
