--- fail2ban-0.9.5/setup.py.~1~	2016-07-15 03:25:07.000000000 +0100
+++ fail2ban-0.9.5/setup.py	2016-09-02 11:55:30.359722588 +0100
@@ -126,29 +126,29 @@
 				for f in w[2]]
 	},
 	data_files = [
-		('/etc/fail2ban',
+		('etc/fail2ban',
 			glob("config/*.conf")
 		),
-		('/etc/fail2ban/filter.d',
+		('etc/fail2ban/filter.d',
 			glob("config/filter.d/*.conf")
 		),
-		('/etc/fail2ban/filter.d/ignorecommands',
+		('etc/fail2ban/filter.d/ignorecommands',
 			glob("config/filter.d/ignorecommands/*")
 		),
-		('/etc/fail2ban/action.d',
+		('etc/fail2ban/action.d',
 			glob("config/action.d/*.conf") +
 			glob("config/action.d/*.py")
 		),
-		('/etc/fail2ban/fail2ban.d',
+		('etc/fail2ban/fail2ban.d',
 			''
 		),
-		('/etc/fail2ban/jail.d',
+		('etc/fail2ban/jail.d',
 			''
 		),
-		('/var/lib/fail2ban',
+		('var/lib/fail2ban',
 			''
 		),
-		('/usr/share/doc/fail2ban',
+		('share/doc/fail2ban',
 			['README.md', 'README.Solaris', 'DEVELOP', 'FILTERS',
 			 'doc/run-rootless.txt']
 		)
