diff -ruN httpd-2.2.17.orig/Makefile.in httpd-2.2.17/Makefile.in
--- httpd-2.2.17.orig/Makefile.in	2008-11-25 12:24:49.000000000 -0800
+++ httpd-2.2.17/Makefile.in	2011-04-01 08:59:35.481942863 -0700
@@ -121,9 +121,10 @@
 	    done ; \
 	done ; \
 	if test -f "$(builddir)/envvars-std"; then \
-	    cp -p envvars-std $(DESTDIR)$(sbindir); \
-	    if test ! -f $(DESTDIR)$(sbindir)/envvars; then \
-	        cp -p envvars-std $(DESTDIR)$(sbindir)/envvars ; \
+	    [ -d $(DESTDIR)$(sysconfdir) ] || mkdir -p $(DESTDIR)$(sysconfdir); \
+	    cp -p envvars-std $(DESTDIR)$(sysconfdir); \
+	    if test ! -f $(DESTDIR)$(sysconfdir)/envvars; then \
+	        cp -p envvars-std $(DESTDIR)$(sysconfdir)/envvars ; \
 	    fi ; \
 	fi
 
diff -ruN httpd-2.2.17.orig/support/Makefile.in httpd-2.2.17/support/Makefile.in
--- httpd-2.2.17.orig/support/Makefile.in	2005-07-06 16:15:34.000000000 -0700
+++ httpd-2.2.17/support/Makefile.in	2011-04-01 08:59:52.202422895 -0700
@@ -30,9 +30,10 @@
 	    fi ; \
 	done
 	@if test -f "$(builddir)/envvars-std"; then \
-	    cp -p envvars-std $(DESTDIR)$(sbindir); \
-	    if test ! -f $(DESTDIR)$(sbindir)/envvars; then \
-	        cp -p envvars-std $(DESTDIR)$(sbindir)/envvars ; \
+	    [ -d $(DESTDIR)$(sysconfdir) ] || mkdir -p $(DESTDIR)$(sysconfdir); \
+	    cp -p envvars-std $(DESTDIR)$(sysconfdir); \
+	    if test ! -f $(DESTDIR)$(sysconfdir)/envvars; then \
+	        cp -p envvars-std $(DESTDIR)$(sysconfdir)/envvars ; \
 	    fi ; \
 	fi
 
