Index: coreutils-8.23/src/fs.h
===================================================================
--- coreutils-8.23/src/fs.h
+++ coreutils-8.23/src/fs.h
@@ -2,7 +2,7 @@
    Please send additions to bug-coreutils@gnu.org and meskes@debian.org.
    This file is generated automatically from ./src/stat.c. */
 
-#if defined __linux__
+#if defined(__linux__) || defined(__sun)
 # define S_MAGIC_ADFS 0xADF5
 # define S_MAGIC_AFFS 0xADFF
 # define S_MAGIC_AFS 0x5346414F
