diff -ruN node_exporter-0.16.0.orig/collector/loadavg_solaris.go node_exporter-0.16.0/collector/loadavg_solaris.go
--- node_exporter-0.16.0.orig/collector/loadavg_solaris.go	2018-05-15 15:49:24.000000000 +0000
+++ node_exporter-0.16.0/collector/loadavg_solaris.go	2018-11-28 13:52:29.305363448 +0000
@@ -21,7 +21,7 @@
 
 /*
 // Define "__stack_chk_fail" and "__stack_chk_guard" symbols.
-#cgo LDFLAGS: -fno-stack-protector -lssp
+#cgo LDFLAGS: -lssp
 // Ensure "hrtime_t" is defined for sys/loadavg.h
 #include <sys/time.h>
 #include <sys/loadavg.h>
