diff -ruN node_exporter-e7664852867694265b093f79a8d8a4dcf27d0e82.orig/Makefile.common node_exporter-e7664852867694265b093f79a8d8a4dcf27d0e82/Makefile.common
--- node_exporter-e7664852867694265b093f79a8d8a4dcf27d0e82.orig/Makefile.common	2019-01-12 12:33:56.000000000 +0000
+++ node_exporter-e7664852867694265b093f79a8d8a4dcf27d0e82/Makefile.common	2019-01-18 08:07:18.036045493 +0000
@@ -195,11 +195,12 @@
 promu: $(PROMU)
 
 $(PROMU):
-	$(eval PROMU_TMP := $(shell mktemp -d))
-	curl -s -L $(PROMU_URL) | tar -xvzf - -C $(PROMU_TMP)
+#	$(eval PROMU_TMP := $(shell mktemp -d))
+#	curl -s -L $(PROMU_URL) | tar -xvzf - -C $(PROMU_TMP)
 	mkdir -p $(FIRST_GOPATH)/bin
-	cp $(PROMU_TMP)/promu-$(PROMU_VERSION).$(GO_BUILD_PLATFORM)/promu $(FIRST_GOPATH)/bin/promu
-	rm -r $(PROMU_TMP)
+#	cp $(PROMU_TMP)/promu-$(PROMU_VERSION).$(GO_BUILD_PLATFORM)/promu $(FIRST_GOPATH)/bin/promu
+	cp /ec/bin/amd64/promu $(FIRST_GOPATH)/bin/promu
+#	rm -r $(PROMU_TMP)
 
 .PHONY: proto
 proto:
