diff -ur apache-cassandra-2.0.4-orig/bin/cassandra apache-cassandra-2.0.4/bin/cassandra
--- apache-cassandra-2.0.4-orig/bin/cassandra	2014-01-21 12:59:28.313545672 +0000
+++ apache-cassandra-2.0.4/bin/cassandra	2014-01-21 13:05:25.599347303 +0000
@@ -69,6 +69,7 @@
     # Locations (in order) to use when searching for an include file.
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/cassandra-cli apache-cassandra-2.0.4/bin/cassandra-cli
--- apache-cassandra-2.0.4-orig/bin/cassandra-cli	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/cassandra-cli	2014-01-21 13:05:37.998250502 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/cassandra-shuffle apache-cassandra-2.0.4/bin/cassandra-shuffle
--- apache-cassandra-2.0.4-orig/bin/cassandra-shuffle	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/cassandra-shuffle	2014-01-21 13:06:10.862927183 +0000
@@ -16,7 +16,8 @@
 # limitations under the License.
 
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
-    for include in /usr/share/cassandra/cassandra.in.sh \
+    for include in /ec/share/cassandra/cassandra.in.sh \
+                   /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh \
                    "`dirname "$0"`/cassandra.in.sh"; do
diff -ur apache-cassandra-2.0.4-orig/bin/cassandra.in.sh apache-cassandra-2.0.4/bin/cassandra.in.sh
--- apache-cassandra-2.0.4-orig/bin/cassandra.in.sh	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/cassandra.in.sh	2014-01-21 13:07:54.719271755 +0000
@@ -15,28 +15,19 @@
 # limitations under the License.
 
 if [ "x$CASSANDRA_HOME" = "x" ]; then
-    CASSANDRA_HOME="`dirname "$0"`/.."
+    CASSANDRA_HOME=/ec/share/cassandra
 fi
 
 # The directory where Cassandra's configs live (required)
 if [ "x$CASSANDRA_CONF" = "x" ]; then
-    CASSANDRA_CONF="$CASSANDRA_HOME/conf"
+    CASSANDRA_CONF=/ec/etc/cassandra
 fi
 
-# This can be the path to a jar file, or a directory containing the 
-# compiled classes. NOTE: This isn't needed by the startup script,
-# it's just used here in constructing the classpath.
-cassandra_bin="$CASSANDRA_HOME/build/classes/main"
-cassandra_bin="$cassandra_bin:$CASSANDRA_HOME/build/classes/thrift"
-#cassandra_bin="$cassandra_home/build/cassandra.jar"
-
-# JAVA_HOME can optionally be set here
-#JAVA_HOME=/usr/local/jdk6
-
 # The java classpath (required)
-CLASSPATH="$CASSANDRA_CONF:$cassandra_bin"
+CLASSPATH="$CASSANDRA_CONF"
 
-for jar in "$CASSANDRA_HOME"/lib/*.jar; do
+for jar in "$CASSANDRA_HOME"/lib/*.jar $CASSANDRA_HOME/*.jar \
+  /ec/share/java/jna-*.jar /ec/share/java/snappy-java-*.jar; do
     CLASSPATH="$CLASSPATH:$jar"
 done
 
@@ -45,4 +36,4 @@
       || [ "$JVM_VERSION" = "1.6.0" -a "$JVM_PATCH_VERSION" -ge 23 ]
 then
     JAVA_AGENT="$JAVA_AGENT -javaagent:$CASSANDRA_HOME/lib/jamm-0.2.5.jar"
-fi
\ No newline at end of file
+fi
diff -ur apache-cassandra-2.0.4-orig/bin/debug-cql apache-cassandra-2.0.4/bin/debug-cql
--- apache-cassandra-2.0.4-orig/bin/debug-cql	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/debug-cql	2014-01-21 13:12:51.366777563 +0000
@@ -17,7 +17,8 @@
 
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     # Locations (in order) to use when searching for an include file.
-    for include in /usr/share/cassandra/cassandra.in.sh \
+    for include in /ec/share/cassandra/cassandra.in.sh \
+                   /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh \
                    "$HOME/.cassandra.in.sh" \
diff -ur apache-cassandra-2.0.4-orig/bin/json2sstable apache-cassandra-2.0.4/bin/json2sstable
--- apache-cassandra-2.0.4-orig/bin/json2sstable	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/json2sstable	2014-01-21 13:03:58.501642658 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/nodetool apache-cassandra-2.0.4/bin/nodetool
--- apache-cassandra-2.0.4-orig/bin/nodetool	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/nodetool	2014-01-21 13:08:19.365421332 +0000
@@ -24,6 +24,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/sstable2json apache-cassandra-2.0.4/bin/sstable2json
--- apache-cassandra-2.0.4-orig/bin/sstable2json	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstable2json	2014-01-21 13:08:37.290769268 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/sstablekeys apache-cassandra-2.0.4/bin/sstablekeys
--- apache-cassandra-2.0.4-orig/bin/sstablekeys	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstablekeys	2014-01-21 13:08:51.075441340 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/sstableloader apache-cassandra-2.0.4/bin/sstableloader
--- apache-cassandra-2.0.4-orig/bin/sstableloader	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstableloader	2014-01-21 13:09:03.943031502 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname "$0"`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/bin/sstablescrub apache-cassandra-2.0.4/bin/sstablescrub
--- apache-cassandra-2.0.4-orig/bin/sstablescrub	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstablescrub	2014-01-21 13:10:04.558999354 +0000
@@ -17,7 +17,8 @@
 # limitations under the License.
 
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
-    for include in /usr/share/cassandra/cassandra.in.sh \
+    for include in /ec/share/cassandra/cassandra.in.sh \
+                   /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh \
                    ~/.cassandra.in.sh \
diff -ur apache-cassandra-2.0.4-orig/bin/sstablesplit apache-cassandra-2.0.4/bin/sstablesplit
--- apache-cassandra-2.0.4-orig/bin/sstablesplit	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstablesplit	2014-01-21 13:09:43.084635500 +0000
@@ -17,7 +17,8 @@
 # limitations under the License.
 
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
-    for include in /usr/share/cassandra/cassandra.in.sh \
+    for include in /ec/share/cassandra/cassandra.in.sh \
+                   /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh \
                    ~/.cassandra.in.sh \
diff -ur apache-cassandra-2.0.4-orig/bin/sstableupgrade apache-cassandra-2.0.4/bin/sstableupgrade
--- apache-cassandra-2.0.4-orig/bin/sstableupgrade	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/bin/sstableupgrade	2014-01-21 13:09:21.366285113 +0000
@@ -17,7 +17,8 @@
 # limitations under the License.
 
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
-    for include in /usr/share/cassandra/cassandra.in.sh \
+    for include in /ec/share/cassandra/cassandra.in.sh \
+                   /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh \
                    ~/.cassandra.in.sh \
diff -ur apache-cassandra-2.0.4-orig/conf/cassandra.yaml apache-cassandra-2.0.4/conf/cassandra.yaml
--- apache-cassandra-2.0.4-orig/conf/cassandra.yaml	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/conf/cassandra.yaml	2014-01-21 13:15:04.109627317 +0000
@@ -103,10 +103,10 @@
 # will spread data evenly across them, subject to the granularity of
 # the configured compaction strategy.
 data_file_directories:
-    - /var/lib/cassandra/data
+    - /ec/var/lib/cassandra/data
 
 # commit log
-commitlog_directory: /var/lib/cassandra/commitlog
+commitlog_directory: /ec/var/lib/cassandra/commitlog
 
 # policy for data disk failures:
 # stop: shut down gossip and Thrift, leaving the node effectively dead, but
@@ -181,7 +181,7 @@
 # memory_allocator: NativeAllocator
 
 # saved caches
-saved_caches_directory: /var/lib/cassandra/saved_caches
+saved_caches_directory: /ec/var/lib/cassandra/saved_caches
 
 # commitlog_sync may be either "periodic" or "batch." 
 # When in batch mode, Cassandra won't ack writes until the commit log
diff -ur apache-cassandra-2.0.4-orig/conf/log4j-server.properties apache-cassandra-2.0.4/conf/log4j-server.properties
--- apache-cassandra-2.0.4-orig/conf/log4j-server.properties	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/conf/log4j-server.properties	2014-01-21 13:15:54.232316900 +0000
@@ -32,7 +32,7 @@
 log4j.appender.R.layout=org.apache.log4j.PatternLayout
 log4j.appender.R.layout.ConversionPattern=%5p [%t] %d{ISO8601} %F (line %L) %m%n
 # Edit the next line to point to your logs directory
-log4j.appender.R.File=/var/log/cassandra/system.log
+log4j.appender.R.File=/ec/var/log/cassandra/system.log
 
 # Application logging options
 #log4j.logger.org.apache.cassandra=DEBUG
diff -ur apache-cassandra-2.0.4-orig/tools/bin/cassandra-stress apache-cassandra-2.0.4/tools/bin/cassandra-stress
--- apache-cassandra-2.0.4-orig/tools/bin/cassandra-stress	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/tools/bin/cassandra-stress	2014-01-21 13:11:50.552991358 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname $0`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
diff -ur apache-cassandra-2.0.4-orig/tools/bin/sstablelevelreset apache-cassandra-2.0.4/tools/bin/sstablelevelreset
--- apache-cassandra-2.0.4-orig/tools/bin/sstablelevelreset	2013-12-27 18:53:08.000000000 +0000
+++ apache-cassandra-2.0.4/tools/bin/sstablelevelreset	2014-01-21 13:11:30.445353287 +0000
@@ -19,6 +19,7 @@
 if [ "x$CASSANDRA_INCLUDE" = "x" ]; then
     for include in "`dirname $0`/cassandra.in.sh" \
                    "$HOME/.cassandra.in.sh" \
+                   /ec/share/cassandra/cassandra.in.sh \
                    /usr/share/cassandra/cassandra.in.sh \
                    /usr/local/share/cassandra/cassandra.in.sh \
                    /opt/cassandra/cassandra.in.sh; do
