--- a/src/corelib/animation/qvariantanimation.cpp.orig    2010-11-06 01:55:18.000000000 +0000
+++ b/src/corelib/animation/qvariantanimation.cpp 2011-01-23 03:58:55.565275558 +0000
@@ -431,12 +431,17 @@
 {
     // will override any existing interpolators
     QInterpolatorVector *interpolators = registeredInterpolators();
+    // When built on solaris with GCC, the destructors can be called
+    // in such an order that we get here with interpolators == NULL,
+    // to continue causes the app to crash on exit with a SEGV
+    if (interpolators) {
 #ifndef QT_NO_THREAD
-    QMutexLocker locker(QMutexPool::globalInstanceGet(interpolators));
+        QMutexLocker locker(QMutexPool::globalInstanceGet(interpolators));
 #endif
-    if (int(interpolationType) >= interpolators->count())
-        interpolators->resize(int(interpolationType) + 1);
-    interpolators->replace(interpolationType, func);
+        if (int(interpolationType) >= interpolators->count())
+            interpolators->resize(int(interpolationType) + 1);
+        interpolators->replace(interpolationType, func);
+    }
 }
 
 
