# https://bugs.webkit.org/show_bug.cgi?id=36508
--- qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/WebCore.pro.orig	2011-11-25 10:16:05.219086146 +0000
+++ qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/WebCore.pro	2011-11-25 10:16:19.307337883 +0000
@@ -89,8 +89,8 @@
     QMAKE_PKGCONFIG_REQUIRES = QtCore QtGui QtNetwork
 }
 
-unix:!mac:*-g++*:QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections 
-unix:!mac:*-g++*:QMAKE_LFLAGS += -Wl,--gc-sections
+unix:!mac:*-g++*:!solaris*:QMAKE_CXXFLAGS += -ffunction-sections -fdata-sections
+unix:!mac:*-g++*:!solaris*:QMAKE_LFLAGS += -Wl,--gc-sections
 linux*-g++*:QMAKE_LFLAGS += $$QMAKE_LFLAGS_NOUNDEF
 
 CONFIG(release):!CONFIG(standalone_package) {
