--- qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/platform/network/ResourceResponseBase.h.orig	2011-11-25 10:32:21.574335143 +0000
+++ qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/platform/network/ResourceResponseBase.h	2011-11-25 10:32:43.999410545 +0000
@@ -32,6 +32,10 @@
 
 #include <wtf/PassOwnPtr.h>
 
+#ifdef OS(SOLARIS)
+#include <sys/time.h>
+#endif
+
 namespace WebCore {
 
 class ResourceResponse;
