--- qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/page/Page.h.orig	2011-11-25 10:31:07.982666082 +0000
+++ qt-everywhere-opensource-src-4.7.3/src/3rdparty/webkit/WebCore/page/Page.h	2011-11-25 10:31:59.674303442 +0000
@@ -26,6 +26,9 @@
 #include <wtf/Forward.h>
 #include <wtf/HashSet.h>
 #include <wtf/Noncopyable.h>
+#ifdef OS(SOLARIS)
+#include <sys/time.h>
+#endif
 
 #if PLATFORM(MAC)
 #include "SchedulePair.h"
