--- x264-snapshot-20110614-2245/configure.orig	2011-06-15 23:09:25.980885451 +0100
+++ x264-snapshot-20110614-2245/configure	2011-06-15 23:09:57.470765915 +0100
@@ -1018,7 +1018,7 @@
     elif [ "$SYS" = "SunOS" ]; then
         echo "SOSUFFIX=so" >> config.mak
         echo "SONAME=libx264.so.$API" >> config.mak
-        echo 'SOFLAGS=-shared -Wl,-h,$(SONAME)' >> config.mak
+        echo 'SOFLAGS=-shared -Wl,-h,$(SONAME) -Wl,-Bsymbolic' >> config.mak
     else
         echo "SOSUFFIX=so" >> config.mak
         echo "SONAME=libx264.so.$API" >> config.mak
