diff -ruN vsftpd-2.3.5.orig/Makefile vsftpd-2.3.5/Makefile
--- vsftpd-2.3.5.orig/Makefile	2009-05-22 19:44:52.000000000 +0000
+++ vsftpd-2.3.5/Makefile	2012-03-06 18:02:53.728101094 +0000
@@ -1,7 +1,7 @@
 # Makefile for systems with GNU tools
 CC 	=	gcc
 INSTALL	=	install
-IFLAGS  = -idirafter dummyinc
+IFLAGS  = $(CPPFLAGS) -idirafter dummyinc
 #CFLAGS = -g
 CFLAGS	=	-O2 -Wall -W -Wshadow #-pedantic -Werror -Wconversion
 
