diff -ruN php-5.3.24.orig/main/main.c php-5.3.24/main/main.c
--- php-5.3.24.orig/main/main.c	2013-04-10 07:42:48.000000000 +0000
+++ php-5.3.24/main/main.c	2013-04-30 14:35:58.011237663 +0000
@@ -1237,7 +1237,7 @@
 		/* can we mmap immeadiately? */
 		memset(&handle->handle.stream.mmap, 0, sizeof(handle->handle.stream.mmap));
 		len = php_zend_stream_fsizer(stream TSRMLS_CC);
-		if (len != 0
+		if (0 && len != 0
 #if HAVE_MMAP || defined(PHP_WIN32)
 		&& ((len - 1) % page_size) <= page_size - ZEND_MMAP_AHEAD
 #endif
