--- vlc-2.1.4/configure.~1~	2014-02-21 14:21:26.000000000 +0000
+++ vlc-2.1.4/configure	2014-04-04 10:39:40.894728086 +0100
@@ -9030,7 +9030,7 @@
   OBJCFLAGS="${OBJCFLAGS} -I${CONTRIB_DIR}/include"
   if test "$build" = "$host" -o "$PKG_CONFIG_LIBDIR"; then :
 
-    export PKG_CONFIG_PATH="${CONTRIB_DIR}/lib/pkgconfig:$PKG_CONFIG_PATH"
+    export PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
 
 else
 
@@ -50927,7 +50927,7 @@
 
 
 cat >>confdefs.h <<_ACEOF
-#define VLC_COMPILE_HOST "`hostname -f 2>/dev/null || hostname`"
+#define VLC_COMPILE_HOST "`hostname 2>/dev/null || hostname`"
 _ACEOF
 
 
