--- vlc-2.0.1/include/vlc_common.h.orig	2012-06-07 00:55:55.990912464 +0000
+++ vlc-2.0.1/include/vlc_common.h	2012-06-07 00:57:49.678184842 +0000
@@ -930,7 +930,7 @@
     if (ptr)
         free((char*)ptr - ((char*)ptr)[-1]);
 }
-#elif defined(__ANDROID__)
+#elif defined(__ANDROID__) || defined (__SVR4) && defined (__sun)
 # define vlc_memalign(align, size) memalign(align, size)
 # define vlc_free(base) free(base)
 #else
