--- vlc-2.0.1/bin/vlc.c.~1~	2011-12-22 10:38:28.000000000 +0000
+++ vlc-2.0.1/bin/vlc.c	2012-06-06 18:49:23.274500361 +0000
@@ -40,10 +40,14 @@
 #endif
 #include <unistd.h>
 
-#ifdef __APPLE__
+#if defined (__APPLE__) || defined (__SVR4) && defined (__sun)
 #include <string.h>
 #endif
 
+#if defined (__SVR4) && defined (__sun)
+#include <sys/mman.h>
+#endif
+
 #ifdef __OS2__
 # define pthread_t      int
 # define pthread_self() _gettid()
@@ -187,6 +191,17 @@
     /* Block all these signals */
     pthread_sigmask (SIG_SETMASK, &set, NULL);
 
+#if defined (__SVR4) && defined (__sun)
+    /* try to fix null strings if glibc doesn't
+       this can't work on latest Linux & BSD because of exploit fix
+    */
+    int page_zero_mapped = ( mmap(0, sysconf( _SC_PAGESIZE ),
+                           PROT_WRITE|PROT_READ,
+                            MAP_FIXED|MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) == 0 );
+    if ( page_zero_mapped )
+      memcpy((char *)0, "(null)\0", 7);
+#endif
+
     /* Note that FromLocale() can be used before libvlc is initialized */
     const char *argv[i_argc + 3];
     int argc = 0;
@@ -262,5 +277,9 @@
     for (int i = 2; i < argc; i++)
         LocaleFree (argv[i]);
 
+#if defined (__SVR4) && defined (__sun)
+    if ( page_zero_mapped ) munmap( 0, sysconf( _SC_PAGESIZE ) );
+#endif
+
     return 0;
 }
