--- pth-2.0.7/Makefile.in.~1~	2006-06-08 17:54:01.000000000 +0000
+++ pth-2.0.7/Makefile.in	2012-09-21 09:41:28.949671055 +0000
@@ -168,10 +168,10 @@
 
 #   build the static and possibly shared libraries
 libpth.la: $(LOBJS)
-	$(LIBTOOL) --mode=link --quiet $(CC) -o libpth.la $(LOBJS) \
+	$(LIBTOOL) --mode=link --quiet $(CC) $(LDFLAGS) -o libpth.la $(LOBJS) $(LIBS) \
 	-rpath $(libdir) -version-info `$(SHTOOL) version -lc -dlibtool $(_VERSION_FILE)`
 libpthread.la: pthread.lo $(LOBJS)
-	$(LIBTOOL) --mode=link --quiet $(CC) -o libpthread.la pthread.lo $(LOBJS) \
+	$(LIBTOOL) --mode=link --quiet $(CC) $(LDFLAGS) -o libpthread.la pthread.lo $(LOBJS) $(LIBS) \
 	-rpath $(libdir) -version-info `$(SHTOOL) version -lc -dlibtool $(_VERSION_FILE)`
 
 #   build the manual pages
