Index: vlc-2.2.1/configure
===================================================================
--- vlc-2.2.1/configure.orig
+++ vlc-2.2.1/configure
@@ -9129,8 +9129,8 @@ $as_echo "${CONTRIB_DIR}" >&6; }
   CFLAGS="${CFLAGS} -I${CONTRIB_DIR}/include"
   CXXFLAGS="${CXXFLAGS} -I${CONTRIB_DIR}/include"
   OBJCFLAGS="${OBJCFLAGS} -I${CONTRIB_DIR}/include"
-  export PKG_CONFIG_PATH="${CONTRIB_DIR}/lib/pkgconfig:$PKG_CONFIG_PATH"
-  LDFLAGS="${LDFLAGS} -L${CONTRIB_DIR}/lib"
+  export PKG_CONFIG_PATH="$PKG_CONFIG_PATH"
+  LDFLAGS="${LDFLAGS}"
 
   if test "${SYS}" = "darwin"; then :
 
@@ -53213,7 +53213,7 @@ _ACEOF
 
 
 cat >>confdefs.h <<_ACEOF
-#define VLC_COMPILE_HOST "`hostname -f 2>/dev/null || hostname`"
+#define VLC_COMPILE_HOST "`hostname 2>/dev/null || hostname`"
 _ACEOF
 
 
