--- php-7.1.22.orig/ext/sockets/config.m4	2018-09-11 14:08:36.000000000 +0000
+++ php-7.1.22/ext/sockets/config.m4	2018-09-18 10:27:43.308518237 +0000
@@ -56,6 +56,6 @@
     AC_DEFINE(HAVE_AI_V4MAPPED,1,[Whether you have AI_V4MAPPED])
   fi
 
-  PHP_NEW_EXTENSION([sockets], [sockets.c multicast.c conversions.c sockaddr_conv.c sendrecvmsg.c], [$ext_shared],, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+  PHP_NEW_EXTENSION([sockets], [sockets.c multicast.c conversions.c sockaddr_conv.c sendrecvmsg.c], [$ext_shared],, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1 -D_XPG4_2)
   PHP_INSTALL_HEADERS([ext/sockets/], [php_sockets.h])
 fi
