--- php-7.2.16/ext/sockets/config.m4.orig	2019-03-25 12:51:18.454586956 +0000
+++ php-7.2.16/ext/sockets/config.m4	2019-03-25 12:50:50.691535801 +0000
@@ -90,6 +90,6 @@
     [sockets],
     [sockets.c multicast.c conversions.c sockaddr_conv.c sendrecvmsg.c],
     [$ext_shared],,
-    $PHP_SOCKETS_CFLAGS)
+    $PHP_SOCKETS_CFLAGS -D_XPG4_2)
   PHP_INSTALL_HEADERS([ext/sockets/], [php_sockets.h])
 fi
