diff -ruN php-7.2.12.orig/ext/intl/config.m4 php-7.2.12/ext/intl/config.m4
--- php-7.2.12.orig/ext/intl/config.m4	2018-11-06 10:48:42.000000000 +0000
+++ php-7.2.12/ext/intl/config.m4	2018-11-14 13:36:37.926580805 +0000
@@ -89,7 +89,7 @@
     breakiterator/rulebasedbreakiterator_methods.cpp \
     breakiterator/codepointiterator_internal.cpp \
     breakiterator/codepointiterator_methods.cpp"
-  PHP_INTL_CXX_FLAGS="$INTL_COMMON_FLAGS $ICU_CXXFLAGS"
+  PHP_INTL_CXX_FLAGS="$INTL_COMMON_FLAGS"
   if test "$ext_shared" = "no"; then
     PHP_ADD_SOURCES(PHP_EXT_DIR(intl), $PHP_INTL_CXX_SOURCES, $PHP_INTL_CXX_FLAGS)
   else
