Index: gcc-4.4.4/gcc/Makefile.in
===================================================================
--- gcc-4.4.4/gcc/Makefile.in
+++ gcc-4.4.4/gcc/Makefile.in
@@ -4725,7 +4725,7 @@ update-po: $(CATALOGS:.gmo=.pox)
 # script does that.
 .po.gmo:
 	$(mkinstalldirs) po
-	$(GMSGFMT) --statistics -o $@ $<
+	$(GMSGFMT) -o $@ $<
 
 # The new .po has to be gone over by hand, so we deposit it into
 # build/po with a different extension.
