--- gcc-4.6.2/gcc/config/i386/sol2-10.h	2011-12-11 00:34:08.341738656 +0100
+++ gcc-4.6.2/gcc/config/i386/sol2-10.h	2011-12-11 00:32:27.545153224 +0100
@@ -131,6 +131,11 @@
 #define LINK_ARCH_SPEC TARGET_LD_EMULATION \
 		       "%{m64:" LINK_ARCH64_SPEC "}%{!m64:" LINK_ARCH32_SPEC "}"
 
+#undef LINK_LIBGCC_SPEC
+#define LINK_LIBGCC_SPEC "%{m64:" "-R /ec/lib/64 %D" "}%{!m64:" "-R /ec/lib %D" "}"
+#undef MD_EXEC_PREFIX
+#define MD_EXEC_PREFIX "/ec/bin/"
+#define LINK_ELIMINATE_DUPLICATE_LDIRECTORIES 0
 /* We do not need to search a special directory for startup files.  */
 #undef MD_STARTFILE_PREFIX
 
