--- gcc-4.4.5/gcc/config/i386/sol2-10.h	2010-07-15 10:08:56.000000000 +0000
+++ gcc-richlowe/gcc/config/i386/sol2-10.h	2011-12-17 01:45:30.033468078 +0000
@@ -94,7 +94,7 @@
    %{R*} \
    %{compat-bsd: \
      %{!YP,*:%{p|pg:-Y P,/usr/ucblib/64:/usr/lib/libp/64:/lib/64:/usr/lib/64} \
-             %{!p:%{!pg:-Y P,/usr/ucblib/64:/lib:/usr/lib/64}}} \
+             %{!p:%{!pg:-Y P,/usr/ucblib/64:/lib/64:/usr/lib/64}}} \
              -R /usr/ucblib/64} \
    %{!compat-bsd: \
      %{!YP,*:%{p|pg:-Y P,/usr/lib/libp/64:/lib/64:/usr/lib/64} \
