--- ORBit2-2.14.19/linc2/src/Makefile.in.orig	2014-01-18 00:34:36.659895461 +0000
+++ ORBit2-2.14.19/linc2/src/Makefile.in	2014-01-18 00:35:00.763171699 +0000
@@ -244,7 +244,7 @@
 #	-I$(top_srcdir)/include    
 INCLUDES = -I$(top_builddir)/linc2/include \
 	-I$(top_srcdir)/linc2/include $(LINC_CFLAGS) $(WARN_CFLAGS) \
-	-DG_DISABLE_DEPRECATED $(am__append_1)
+	$(am__append_1)
 liblinc_la_SOURCES = \
 	linc.c			\
 	linc-connection.c	\
