Index: libtool-2.4.2/Makefile.in
===================================================================
--- libtool-2.4.2/Makefile.in
+++ libtool-2.4.2/Makefile.in
@@ -533,7 +533,7 @@ timestamp = set dummy `$(MKSTAMP) $(srcd
 	  *) TIMESTAMP="" ;; \
 	esac
 
-rebuild = rebuild=:; $(timestamp); correctver=$$1
+rebuild = rebuild=:; $(timestamp); correctver=1.3337
 
 # ---------- #
 # Bootstrap. #
Index: libtool-2.4.2/Makefile.am
===================================================================
--- libtool-2.4.2/Makefile.am
+++ libtool-2.4.2/Makefile.am
@@ -57,7 +57,7 @@ timestamp = set dummy `$(MKSTAMP) $(srcd
 	  *) TIMESTAMP="" ;; \
 	esac
 
-rebuild = rebuild=:; $(timestamp); correctver=$$1
+rebuild = rebuild=:; $(timestamp); correctver=1.3337
 
 
 # ---------- #
