--- cairo-1.12.16/configure.orig	2014-01-17 12:56:47.517693228 +0000
+++ cairo-1.12.16/configure	2014-01-17 13:02:22.244210842 +0000
@@ -30904,7 +30904,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_POPPLER_CFLAGS=`$PKG_CONFIG --cflags "$poppler_DEPENDENCY" 2>/dev/null`
+  pkg_cv_POPPLER_CFLAGS=`$PKG_CONFIG --cflags "$poppler_DEPENDENCY" 2>/dev/null | sed 's!-I/ec/include/cairo!!'`
 else
   pkg_failed=yes
 fi
@@ -30920,7 +30920,7 @@
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_POPPLER_LIBS=`$PKG_CONFIG --libs "$poppler_DEPENDENCY" 2>/dev/null`
+  pkg_cv_POPPLER_LIBS=`$PKG_CONFIG --libs "$poppler_DEPENDENCY" 2>/dev/null | sed 's!-lcairo!!'`
 else
   pkg_failed=yes
 fi
