--- nginx-1.2.6/auto/install.orig	2012-12-20 15:15:50.105449161 +0000
+++ nginx-1.2.6/auto/install	2012-12-20 15:17:38.253030245 +0000
@@ -8,7 +8,7 @@
     cat << END                                                >> $NGX_MAKEFILE
 
 install_perl_modules:
-	cd $NGX_OBJS/src/http/modules/perl && \$(MAKE) install
+	cd $NGX_OBJS/src/http/modules/perl && \$(MAKE) pure_install
 END
 
     NGX_INSTALL_PERL_MODULES=install_perl_modules
@@ -147,8 +147,9 @@
 	test -d '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`' || \
 		mkdir -p '\$(DESTDIR)`dirname "$NGX_HTTP_LOG_PATH"`'
 
-	test -d '\$(DESTDIR)$NGX_PREFIX/html' \
-		|| cp -R $NGX_HTML '\$(DESTDIR)$NGX_PREFIX'
+	test -d '\$(DESTDIR)$NGX_PREFIX/share/nginx/html' || \
+		mkdir -p '\$(DESTDIR)$NGX_PREFIX/share/nginx/html' && \
+		cp -R $NGX_HTML '\$(DESTDIR)$NGX_PREFIX/share/nginx'
 END
 
 
