--- pdns-recursor-3.6.1/Makefile.orig	2014-10-10 14:54:04.979823611 +0000
+++ pdns-recursor-3.6.1/Makefile	2014-10-10 14:54:34.242579549 +0000
@@ -3,6 +3,7 @@
 BINDIR=/usr/bin/
 SYSCONFDIR=/etc/powerdns/
 LOCALSTATEDIR=/var/run/
+SHAREDIR=/usr/share
 OPTFLAGS?=-O3
 CXXFLAGS:= $(CXXFLAGS) -Iext/rapidjson/include -I$(CURDIR)/ext/polarssl-1.3.2/include -Wall $(OPTFLAGS) $(PROFILEFLAGS) $(ARCHFLAGS) -pthread -Iext/yahttp
 CFLAGS:=$(CFLAGS) -Wall $(OPTFLAGS) $(PROFILEFLAGS) $(ARCHFLAGS) -I$(CURDIR)/ext/polarssl-1.3.2/include -pthread
@@ -113,8 +114,8 @@
 endif
 	-mkdir -p $(DESTDIR)/$(SYSCONFDIR)
 	$(DESTDIR)/$(SBINDIR)/pdns_recursor --config > $(DESTDIR)/$(SYSCONFDIR)/recursor.conf-dist
-	-mkdir -p $(DESTDIR)/usr/share/man/man1
-	cp pdns_recursor.1 rec_control.1 $(DESTDIR)/usr/share/man/man1
+	-mkdir -p $(DESTDIR)/$(SHAREDIR)/man/man1
+	cp pdns_recursor.1 rec_control.1 $(DESTDIR)/$(SHAREDIR)/man/man1
 	$(OS_SPECIFIC_INSTALL)	
 
 clean: binclean
