--- exim-4.84/src/exim.h.orig	2014-08-23 14:16:06.112683601 +0100
+++ exim-4.84/src/exim.h	2014-08-23 14:16:36.199456395 +0100
@@ -515,7 +515,7 @@
 requires various things that are set therein. */
 
 #if HAVE_ICONV             /* Not all OS have this */
-# include <iconv.h>
+# include </ec/include/iconv.h>
 #endif
 
 #if defined(USE_READLINE) || defined(EXPAND_DLFUNC) || defined (LOOKUP_MODULE_DIR)
