--- mtr/configure.ac	2013-10-25 11:34:35.000000000 +0000
+++ mtr/configure.ac	2013-10-25 11:35:09.827317223 +0000
@@ -23,7 +23,7 @@
 # We don't refer to any symbols in termcap, but -lcurses on SunOS does.
 # We have to trust the linker not to mess things up... (It should not
 # pull in anything if we don't refer to anything in the lib). 
-AC_CHECK_LIB(termcap, tgetent)
+#AC_CHECK_LIB(termcap, tgetent)
 AC_CHECK_LIB(tinfo, tgetent)
 
 AC_CHECK_FUNC(initscr, , 
