--- mysql-5.7.22/rapid/plugin/group_replication/rpcgen.cmake.orig	2018-04-19 15:40:06.324377143 +0000
+++ mysql-5.7.22/rapid/plugin/group_replication/rpcgen.cmake	2018-04-19 15:41:56.908434694 +0000
@@ -64,7 +64,7 @@
   SET (x_vanilla_h      ${XCOM_BASEDIR}/${X}.h.gen)
   SET (x_vanilla_c      ${XCOM_BASEDIR}/${X}_xdr.c.gen)
 
-  IF(WIN32)
+#  IF(WIN32)
     # on windows system's there is no rpcgen, thence copy
     # the files in the source directory
     ADD_CUSTOM_COMMAND(OUTPUT ${x_gen_h} ${x_gen_c} ${x_tmp_plat_h}
@@ -86,7 +86,8 @@
         ${x_vanilla_plat_h}
         ${XCOM_BASEDIR}/xcom_proto_enum.h
         ${XCOM_BASEDIR}/xcom_limits.h)
-  ELSE()
+#  ELSE()
+  IF(0)
     FIND_PROGRAM(RPCGEN_EXECUTABLE rpcgen DOC "path to the rpcgen executable")
     MARK_AS_ADVANCED(RPCGEN_EXECUTABLE)
     IF(NOT RPCGEN_EXECUTABLE)
