--- MediaInfoLib/Source/MediaInfo/MediaInfo_Config.h.orig	2011-10-14 13:50:33.785046419 +0100
+++ MediaInfoLib/Source/MediaInfo/MediaInfo_Config.h	2011-10-14 13:51:45.413052516 +0100
@@ -28,10 +28,10 @@
 
 //---------------------------------------------------------------------------
 #include "MediaInfo/MediaInfo_Internal_Const.h"
-#include "ZenLib/CriticalSection.h"
 #include "ZenLib/ZtringListList.h"
 #include "ZenLib/Translation.h"
 #include "ZenLib/InfoMap.h"
+#include "ZenLib/CriticalSection.h"
 #include <map>
 #include <vector>
 #include <bitset>
