diff --git a/apache2/modsecurity.c b/apache2/modsecurity.c
index da89faa..5bda4cf 100644
--- a/apache2/modsecurity.c
+++ b/apache2/modsecurity.c
@@ -237,8 +237,6 @@ static void modsecurity_persist_data(modsec_rec *msr) {
     }
 
     /* Remove stale collections. */
-    srand(time(NULL));
-
     if (rand() < RAND_MAX/100) {
         arr = apr_table_elts(msr->collections);
         te = (apr_table_entry_t *)arr->elts;
