--- gnome-vfs-2.24.4/configure.in.dist	2013-12-13 14:54:03.861793297 +0000
+++ gnome-vfs-2.24.4/configure.in	2013-12-13 14:54:17.222502115 +0000
@@ -451,8 +451,8 @@
 AC_PATH_PROG([KRB5_CONFIG], krb5-config, none, $PATH:/usr/kerberos/bin)
 					   
 if test "x$KRB5_CONFIG" != "xnone"; then
-   GSSAPI_LIBS="`${KRB5_CONFIG} --libs gssapi`"
-   GSSAPI_CFLAGS="`${KRB5_CONFIG} --cflags gssapi`"
+   GSSAPI_LIBS="`${KRB5_CONFIG} --libs`"
+   GSSAPI_CFLAGS="`${KRB5_CONFIG} --cflags`"
    
    saved_CPPFLAGS="$CPPFLAGS"
    saved_LIBS="$LIBS"
