diff -ruN pkg-gate.orig/src/setup.py pkg-gate/src/setup.py
--- pkg-gate.orig/src/setup.py	2011-12-19 06:54:04.000000000 +0000
+++ pkg-gate/src/setup.py	2011-12-19 10:28:34.083104803 +0000
@@ -496,15 +496,6 @@
 
                 prep_sw(CP, CPARC, CPDIR, CPURL, CPHASH)
                 install_sw(CP, CPDIR, CPIDIR)
-		if osname == "sunos" and platform.uname()[2] == "5.11":
-                        prep_sw(LDTP, LDTPARC, LDTPDIR, LDTPURL,
-                            LDTPHASH)
-                        saveenv = os.environ.copy()
-                        os.environ["LDFLAGS"] = os.environ.get("LDFLAGS", "") + \
-                            " -lsocket -lnsl"
-                        install_ldtp(LDTP, LDTPDIR, LDTPIDIR)
-                        os.environ = saveenv
-
 		if "BUILD_PYOPENSSL" in os.environ and \
                     os.environ["BUILD_PYOPENSSL"] != "":
                         #
