--- pkg-gate/src/modules/flavor/elf.py.orig	2011-12-25 03:01:56.011231205 +0000
+++ pkg-gate/src/modules/flavor/elf.py	2011-12-25 03:02:38.644788937 +0000
@@ -142,7 +142,7 @@
                         res.append(p)
         return res, elist
 
-default_run_paths = ["/lib", "/usr/lib"]
+default_run_paths = ["/lib", "/usr/lib", "/ec/lib"]
 
 def process_elf_dependencies(action, pkg_vars, dyn_tok_conv, run_paths,
     **kwargs):
