diff -urN pkg-gate.orig//src/setup.py pkg-gate//src/setup.py
--- pkg-gate.orig//src/setup.py	2011-03-11 16:49:05.000000000 +0000
+++ pkg-gate//src/setup.py	2011-09-29 14:47:12.745732814 +0100
@@ -144,11 +144,11 @@
 
 M2C = 'M2Crypto'
 M2CIDIR = 'm2crypto'
-M2CVER = '0.20.2'
+M2CVER = '0.21.1'
 M2CARC = '%s-%s.tar.gz' % (M2C, M2CVER)
 M2CDIR = '%s-%s' % (M2C, M2CVER)
 M2CURL = 'http://pypi.python.org/packages/source/M/M2Crypto/%s' % (M2CARC)
-M2CHASH = '6e85d7035971c353db74a497fca0ec2eaea4dea0'
+M2CHASH = '3c7135b952092e4f2eee7a94c5153319cccba94e'
 
 pwd = os.path.normpath(sys.path[0])
 
