diff --git a/src/modules/flavor/python.py b/src/modules/flavor/python.py
--- a/src/modules/flavor/python.py
+++ b/src/modules/flavor/python.py
@@ -124,8 +124,8 @@
                     self.base_names, self.run_paths, self.pkg_vars)
 
 
-py_bin_re = re.compile(r"^\#\!\s*/usr/bin/python(?P<major>\d+)\.(?P<minor>\d+)")
-py_lib_re = re.compile(r"^usr/lib/python(?P<major>\d+)\.(?P<minor>\d+)/")
+py_bin_re = re.compile(r"^\#\!.*/bin/python(?P<major>\d+)\.(?P<minor>\d+)")
+py_lib_re = re.compile(r"^.*/lib/python(?P<major>\d+)\.(?P<minor>\d+)/")
 
 def process_python_dependencies(action, pkg_vars, script_path, run_paths):
         """Analyze the file delivered by the action for any python dependencies.
