--- pkg-gate/src/setup.py.orig	2014-01-02 01:26:01.872600362 +0000
+++ pkg-gate/src/setup.py	2014-01-02 01:28:29.061333293 +0000
@@ -108,11 +108,11 @@
 
 MAKO = 'Mako'
 MAKOIDIR = 'mako'
-MAKOVER = '0.2.2'
+MAKOVER = '0.3.6'
 MAKOARC = '%s-%s.tar.gz' % (MAKO, MAKOVER)
 MAKODIR = '%s-%s' % (MAKO, MAKOVER)
-MAKOURL = 'http://www.makotemplates.org/downloads/%s' % (MAKOARC)
-MAKOHASH = '85c04ab3a6a26a1cab47067449712d15a8b29790'
+MAKOURL = 'https://pypi.python.org/packages/source/M/Mako/%s' % (MAKOARC)
+MAKOHASH = 'ea0df17702107e6498853248ecbe227ead18eb76'
 
 PLY = 'ply'
 PLYIDIR = 'ply'
