diff -r ed0aeb87d2e3 src/Makefile
--- a/src/Makefile	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/Makefile	Thu Mar 10 23:21:54 2011 +0000
@@ -23,7 +23,7 @@
 # Copyright (c) 2007, 2010, Oracle and/or its affiliates. All rights reserved.
 #
 
-PYTHON = /usr/bin/python2.6
+PYTHON = /ec/bin/python2.6
 
 all := TARGET = all
 install := TARGET = install
@@ -35,7 +35,7 @@
 test-generate := TARGET = test-generate
 test-leaks := TARGET = test-leaks
 
-SUBDIRS=web gui um po util/misc brand
+SUBDIRS=
 
 all: $(SUBDIRS)
 	$(PYTHON) setup.py build
diff -r ed0aeb87d2e3 src/client.py
--- a/src/client.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/client.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -4886,7 +4886,7 @@
 
 if __name__ == "__main__":
         misc.setlocale(locale.LC_ALL, "", error)
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         # Make all warnings be errors.
         import warnings
diff -r ed0aeb87d2e3 src/depot.py
--- a/src/depot.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/depot.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -39,7 +39,7 @@
 # dumb clients (like a notification service).
 
 # The default path for static and other web content.
-CONTENT_PATH_DEFAULT = "/usr/share/lib/pkg"
+CONTENT_PATH_DEFAULT = "/ec/share/lib/pkg"
 # cherrypy has a max_request_body_size parameter that determines whether the
 # server should abort requests with REQUEST_ENTITY_TOO_LARGE when the request
 # body is larger than the specified size (in bytes).  The maximum size supported
@@ -124,7 +124,7 @@
                 sys.exit(retcode)
 
         print """\
-Usage: /usr/lib/pkg.depotd [-a address] [-d inst_root] [-p port] [-s threads]
+Usage: /ec/lib/pkg.depotd [-a address] [-d inst_root] [-p port] [-s threads]
            [-t socket_timeout] [--cfg] [--content-root]
            [--disable-ops op[/1][,...]] [--debug feature_list]
            [--file-root dir] [--log-access dest] [--log-errors dest]
@@ -155,7 +155,7 @@
         --content-root  The file system path to the directory containing the
                         the static and other web content used by the depot's
                         browser user interface.  The default value is
-                        '/usr/share/lib/pkg'.
+                        '/ec/share/lib/pkg'.
         --disable-ops   A comma separated list of operations that the depot
                         should not configure.  If, for example, you wanted
                         to omit loading search v1, 'search/1' should be
@@ -228,7 +228,7 @@
 if __name__ == "__main__":
 
         setlocale(locale.LC_ALL, "")
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         add_content = False
         exit_ready = False
diff -r ed0aeb87d2e3 src/pkgdep.py
--- a/src/pkgdep.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/pkgdep.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -495,7 +495,7 @@
 
 def main_func():
         misc.setlocale(locale.LC_ALL, "", error)
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         try:
                 opts, pargs = getopt.getopt(sys.argv[1:], "R:?",
diff -r ed0aeb87d2e3 src/pkgrepo.py
--- a/src/pkgrepo.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/pkgrepo.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -1326,7 +1326,7 @@
 
 if __name__ == "__main__":
         misc.setlocale(locale.LC_ALL, "", error)
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         # Make all warnings be errors.
         warnings.simplefilter('error')
diff -r ed0aeb87d2e3 src/publish.py
--- a/src/publish.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/publish.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -594,7 +594,7 @@
         return xport, targ_pub
 
 def main_func():
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         try:
                 repo_uri = os.environ["PKG_REPO"]
diff -r ed0aeb87d2e3 src/pull.py
--- a/src/pull.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/pull.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -394,7 +394,7 @@
 
         temp_root = misc.config_temp_root()
 
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         global_settings.client_name = "pkgrecv"
         target = os.environ.get("PKG_DEST", None)
diff -r ed0aeb87d2e3 src/setup.py
--- a/src/setup.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/setup.py	Thu Mar 10 23:21:54 2011 +0000
@@ -168,16 +168,16 @@
 pkgs_dir = os.path.normpath(os.path.join(pwd, os.pardir, "packages", arch))
 extern_dir = os.path.normpath(os.path.join(pwd, "extern"))
 
-py_install_dir = 'usr/lib/python2.6/vendor-packages'
+py_install_dir = 'ec/lib/python2.6/vendor-packages'
 
-scripts_dir = 'usr/bin'
-lib_dir = 'usr/lib'
+scripts_dir = 'ec/bin'
+lib_dir = 'ec/lib'
 svc_method_dir = 'lib/svc/method'
 
-man1_dir = 'usr/share/man/cat1'
-man1m_dir = 'usr/share/man/cat1m'
-man5_dir = 'usr/share/man/cat5'
-resource_dir = 'usr/share/lib/pkg'
+man1_dir = 'ec/share/man/cat1'
+man1m_dir = 'ec/share/man/cat1m'
+man5_dir = 'ec/share/man/cat5'
+resource_dir = 'ec/share/lib/pkg'
 smf_dir = 'lib/svc/manifest/application'
 zones_dir = 'etc/zones'
 etcbrand_dir = 'etc/brand/ipkg'
@@ -546,7 +546,7 @@
                         shutil.rmtree(os.path.join(root_dir, py_install_dir, dir),
                             onerror=onerror)
                 try:
-                        os.remove(os.path.join(root_dir, "usr/bin/mako-render"))
+                        os.remove(os.path.join(root_dir, "ec/bin/mako-render"))
                 except EnvironmentError, e:
                         if e.errno != errno.ENOENT:
                                 raise
@@ -611,7 +611,7 @@
                         patchpath = os.path.join(os.path.pardir,
                             os.path.pardir, patchdir, p)
                         print "Applying %s to %s" % (p, swname)
-                        args = ["patch", "-d", swdir, "-i", patchpath, "-p0"]
+                        args = ["gpatch", "-d", swdir, "-i", patchpath, "-p0"]
                         if osname == "windows":
                                 args.append("--binary")
                         ret = subprocess.Popen(args).wait()
diff -r ed0aeb87d2e3 src/sign.py
--- a/src/sign.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/sign.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -123,7 +123,7 @@
 
 def main_func():
         misc.setlocale(locale.LC_ALL, "", error)
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
         global_settings.client_name = "pkgsign"
 
         try:
diff -r ed0aeb87d2e3 src/svc/svc-pkg-depot
--- a/src/svc/svc-pkg-depot	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/svc/svc-pkg-depot	Thu Mar 10 23:21:54 2011 +0000
@@ -52,15 +52,15 @@
 	# adjust the PYTHONPATH to point to the current environment
 	# we need to make sure to adjust the PYTHONPATH accordingly
 	# to a Python 2.4 or 2.6 environment
-	python_ver=$(head -1 ${pkg_root}usr/lib/pkg.depotd 2>/dev/null |
+	python_ver=$(head -1 ${pkg_root}ec/lib/pkg.depotd 2>/dev/null |
 	    awk -F/ '{print $NF}')
 	if [[ $python_ver != *python* ]]; then
 		echo "invalid python version $python_ver found in"
-		echo "${pkg_root}usr/lib/pkg.depotd"
+		echo "${pkg_root}ec/lib/pkg.depotd"
 		exit $SMF_EXIT_ERR_FATAL
 	fi
 
-	PYTHONPATH=${pkg_root}usr/lib/${python_ver}/vendor-packages/:$PYTHONPATH
+	PYTHONPATH=${pkg_root}ec/lib/${python_ver}/vendor-packages/:$PYTHONPATH
 
 	export PYTHONPATH
 
@@ -84,7 +84,7 @@
             A=basic,-file_link_any,-proc_info,-proc_session$privaddr -e"
 
 	# Build the command to start pkg.depotd.
-	cmd="$wrapper ${pkg_root}usr/lib/pkg.depotd --cfg $SMF_FMRI"
+	cmd="$wrapper ${pkg_root}ec/lib/pkg.depotd --cfg $SMF_FMRI"
 
 	# Echo the command so that the log contains the command used to start
 	# the depot.
diff -r ed0aeb87d2e3 src/svc/svc-pkg-mdns
--- a/src/svc/svc-pkg-mdns	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/svc/svc-pkg-mdns	Thu Mar 10 23:21:54 2011 +0000
@@ -61,15 +61,15 @@
 	# adjust the PYTHONPATH to point to the current environment
 	# we need to make sure to adjust the PYTHONPATH accordingly
 	# to a Python 2.4 or 2.6 environment
-	python_ver=$(head -1 ${pkg_root}usr/lib/pkg.depotd 2>/dev/null |
+	python_ver=$(head -1 ${pkg_root}ec/lib/pkg.depotd 2>/dev/null |
 	    awk -F/ '{print $NF}')
 	if [[ $python_ver != *python* ]]; then
 		echo "invalid python version $python_ver found in"
-		echo "${pkg_root}usr/lib/pkg.depotd"
+		echo "${pkg_root}ec/lib/pkg.depotd"
 		exit $SMF_EXIT_ERR_FATAL
 	fi
 
-	PYTHONPATH=${pkg_root}usr/lib/${python_ver}/vendor-packages/:$PYTHONPATH
+	PYTHONPATH=${pkg_root}ec/lib/${python_ver}/vendor-packages/:$PYTHONPATH
 
 	export PYTHONPATH
 
@@ -114,7 +114,7 @@
 	option_ops="$option_ops --llmirror"
 
 	# Build the command to start pkg.depotd with the specified options.
-	cmd="${pkg_root}usr/lib/pkg.depotd $option_ops"
+	cmd="${pkg_root}ec/lib/pkg.depotd $option_ops"
 	# Echo the command so that the log contains the command used to start
 	# the depot.
 	echo $cmd
diff -r ed0aeb87d2e3 src/util/pkglintrc
--- a/src/util/pkglintrc	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/pkglintrc	Thu Mar 10 23:21:54 2011 +0000
@@ -48,10 +48,10 @@
 
 # The version pattern we use when searching for manifests
 # for a given build (only when using the -b flag to pkglint)
-version.pattern = *,5.11-0.
+version.pattern = *,5.10-0.
 
 # The path used for info.classification attribute checks
-info_classification_path = /usr/share/lib/pkg/opensolaris.org.sections
+info_classification_path = /ec/share/lib/pkg/opensolaris.org.sections
 
 # Whether to ignore publisher differences when comparing package versions
 ignore_different_publishers = True
diff -r ed0aeb87d2e3 src/util/publish/pkgdiff.py
--- a/src/util/publish/pkgdiff.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/publish/pkgdiff.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -44,7 +44,7 @@
                 print >> sys.stderr, "pkgdiff: %s" % errmsg
 
         print _(
-            "/usr/bin/pkgdiff [-i attribute ...] [-o attribute] [-v variant=value ...] file1 file2")
+            "/ec/bin/pkgdiff [-i attribute ...] [-o attribute] [-v variant=value ...] file1 file2")
         sys.exit(exitcode)
 
 def error(text, exitcode=1):
@@ -56,7 +56,7 @@
                 sys.exit(exitcode)
 
 def main_func():
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         ignoreattrs = []
         onlyattrs = []
diff -r ed0aeb87d2e3 src/util/publish/pkgfmt.py
--- a/src/util/publish/pkgfmt.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/publish/pkgfmt.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -245,7 +245,7 @@
 
 
 def main_func():
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
         global opt_unwrap
         global opt_check
         global opt_diffs
diff -r ed0aeb87d2e3 src/util/publish/pkglint.py
--- a/src/util/publish/pkglint.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/publish/pkglint.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#! /usr/bin/python
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -32,7 +32,7 @@
 import traceback
 from optparse import OptionParser
 
-gettext.install("pkg", "/usr/share/locale")
+gettext.install("pkg", "/ec/share/locale")
 
 from pkg.client.api_errors import InvalidPackageErrors
 from pkg import VERSION
diff -r ed0aeb87d2e3 src/util/publish/pkgmerge.py
--- a/src/util/publish/pkgmerge.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/publish/pkgmerge.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -864,7 +864,7 @@
 
 if __name__ == "__main__":
         misc.setlocale(locale.LC_ALL, "", error)
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         # Make all warnings be errors.
         import warnings
diff -r ed0aeb87d2e3 src/util/publish/pkgmogrify.py
--- a/src/util/publish/pkgmogrify.py	Thu Mar 10 14:34:53 2011 +0000
+++ b/src/util/publish/pkgmogrify.py	Thu Mar 10 23:21:54 2011 +0000
@@ -1,4 +1,4 @@
-#!/usr/bin/python2.6
+#!/ec/bin/python2.6
 #
 # CDDL HEADER START
 #
@@ -50,7 +50,7 @@
                 print >> sys.stderr, "pkgmogrify: %s" % errmsg
 
         print _(
-            "/usr/bin/pkgmogrify [-vi] [-I includedir ...] [-D macro=value ...] "
+            "/ec/bin/pkgmogrify [-vi] [-I includedir ...] [-D macro=value ...] "
             "[-O outputfile] [-P printfile] [inputfile ...]")
         sys.exit(exitcode)
 
@@ -659,7 +659,7 @@
                 sys.exit(exitcode)
 
 def main_func():
-        gettext.install("pkg", "/usr/share/locale")
+        gettext.install("pkg", "/ec/share/locale")
 
         outfilename = None
         printfilename = None
