--- gettext-0.19.8/gettext-tools/src/msgfmt.c.~1~	2016-05-30 09:12:33.000000000 +0000
+++ gettext-0.19.8/gettext-tools/src/msgfmt.c	2017-01-23 22:25:20.468106374 +0000
@@ -254,6 +254,9 @@
   alignment = DEFAULT_OUTPUT_ALIGNMENT;
   byteswap = 0 ^ ENDIANNESS;
 
+  /* Changed by Debian: Default is little-endian, not native */
+  byteswap = ENDIANNESS;
+
   /* Set program name for messages.  */
   set_program_name (argv[0]);
   error_print_progname = maybe_print_progname;
